--- toolchain/logs/llvm/kepler.SCHWINGE/log_install_	2013-02-09 15:21:22.363356016 +0100
+++ toolchain/logs/llvm/coulomb.SCHWINGE/log_install_	2013-02-09 15:21:22.371356017 +0100
@@ -18,29 +18,28 @@
 	    /usr/bin/install -c -m 0644 $hdr [...]/tschwinge/Horace_Silver.build.install/include/$hdr ; \
 	  done ; \
 	fi
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm-c
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm-c/Transforms
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/ADT
 llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Analysis
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/DebugInfo
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Support
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Transforms
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Transforms/IPO
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Transforms/Utils
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Object
 llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Assembly
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Target
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Option
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/IR
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Bitcode
 llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/CodeGen
 llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/CodeGen/PBQP
 llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/CodeGen/PBQP/Heuristics
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Bitcode
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/DebugInfo
 llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/ExecutionEngine
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/ADT
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/TableGen
 llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/MC
 llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/MC/MCParser
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/IR
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm-c
-llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm-c/Transforms
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Object
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Option
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Support
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/TableGen
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Target
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Transforms
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Transforms/IPO
+llvm[0]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/./llvm/Transforms/Utils
 if test -d "[...]/tschwinge/Horace_Silver.build/include" ; then \
 	  cd [...]/tschwinge/Horace_Silver.build/include && \
 	  for hdr in `find . -type f \
@@ -2143,34 +2142,33 @@
 	    /usr/bin/install -c -m 0644 $hdr [...]/tschwinge/Horace_Silver.build.install/include/$hdr ; \
 	  done ; \
 	fi
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Lex
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Analysis
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Analysis/Visitors
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Analysis/Support
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Analysis/FlowSensitive
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Analysis/DomainSpecific
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Analysis/Analyses
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Format
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Serialization
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Parse
 llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/ARCMigrate
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/FrontendTool
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/AST
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/ASTMatchers
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Analysis/Analyses
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Analysis/DomainSpecific
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Analysis/FlowSensitive
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Analysis/Support
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Analysis/Visitors
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Basic
 llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/CodeGen
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Driver
 llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Edit
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Tooling
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/AST
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Format
 llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Frontend
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/FrontendTool
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Lex
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Parse
 llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Rewrite/Core
 llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Rewrite/Frontend
 llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Sema
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Serialization
 llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/StaticAnalyzer/Checkers
 llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/StaticAnalyzer/Core
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/StaticAnalyzer/Core/PathSensitive
 llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/StaticAnalyzer/Core/BugReporter
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/StaticAnalyzer/Core/PathSensitive
 llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/StaticAnalyzer/Frontend
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Driver
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/ASTMatchers
-llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Basic
+llvm[4]: Making install directory [...]/tschwinge/Horace_Silver.build.install/include/clang/Tooling
 if test -d "[...]/tschwinge/Horace_Silver.build/tools/clang/include/clang" ; then \
 	  cd [...]/tschwinge/Horace_Silver.build/tools/clang/include && \
 	  for hdr in `find clang -type f \
@@ -2851,25 +2849,7 @@
 	fi; \
 	make -C $DD install
 make[4]: Entering directory `[...]/tschwinge/Horace_Silver.build/tools/clang/runtime/compiler-rt'
-make -C [...]/tschwinge/Horace_Silver/projects/compiler-rt \
-	  ProjSrcRoot=[...]/tschwinge/Horace_Silver/projects/compiler-rt \
-	  ProjObjRoot=[...]/tschwinge/Horace_Silver.build/tools/clang/runtime/compiler-rt \
-	  CC="[...]/tschwinge/Horace_Silver.build/Release+Asserts/bin/clang" \
-	  LLVM_ANDROID_TOOLCHAIN_DIR="" \
-	  clang_linux
-make[5]: Entering directory `[...]/compiler-rt/tschwinge/Doug_Watkins'
-make[5]: Nothing to be done for `clang_linux'.
-make[5]: Leaving directory `[...]/compiler-rt/tschwinge/Doug_Watkins'
-[...]/tschwinge/Horace_Silver/autoconf/mkinstalldirs [...]/tschwinge/Horace_Silver.build.install/lib/clang/3.3/lib
-[...]/tschwinge/Horace_Silver/autoconf/mkinstalldirs [...]/tschwinge/Horace_Silver.build.install/lib/clang/3.3/lib/linux
-llvm[4]: Installing compiler runtime library: linux/full-i386
-/usr/bin/install -c -m 0644 [...]/tschwinge/Horace_Silver.build/Release+Asserts/lib/clang/3.3/lib/linux/libclang_rt.full-i386.a [...]/tschwinge/Horace_Silver.build.install/lib/clang/3.3/lib/linux
-llvm[4]: Installing compiler runtime library: linux/profile-i386
-/usr/bin/install -c -m 0644 [...]/tschwinge/Horace_Silver.build/Release+Asserts/lib/clang/3.3/lib/linux/libclang_rt.profile-i386.a [...]/tschwinge/Horace_Silver.build.install/lib/clang/3.3/lib/linux
-llvm[4]: Installing compiler runtime library: linux/asan-i386
-/usr/bin/install -c -m 0644 [...]/tschwinge/Horace_Silver.build/Release+Asserts/lib/clang/3.3/lib/linux/libclang_rt.asan-i386.a [...]/tschwinge/Horace_Silver.build.install/lib/clang/3.3/lib/linux
-llvm[4]: Installing compiler runtime library: linux/ubsan-i386
-/usr/bin/install -c -m 0644 [...]/tschwinge/Horace_Silver.build/Release+Asserts/lib/clang/3.3/lib/linux/libclang_rt.ubsan-i386.a [...]/tschwinge/Horace_Silver.build.install/lib/clang/3.3/lib/linux
+make[4]: Nothing to be done for `install'.
 make[4]: Leaving directory `[...]/tschwinge/Horace_Silver.build/tools/clang/runtime/compiler-rt'
 \
 	  SD=[...]/tschwinge/Horace_Silver/tools/clang/runtime/libcxx; \
