--- toolchain/logs/gdb/kepler.SCHWINGE/install_	2012-11-06 13:03:32.177518466 +0100
+++ toolchain/logs/gdb/coulomb.SCHWINGE/install_	2012-11-06 13:03:32.181518467 +0100
@@ -68,7 +68,6 @@
 libtool: install: /usr/bin/install -c .libs/libbfd.a [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libbfd.a
 libtool: install: chmod 644 [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libbfd.a
 libtool: install: ranlib [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libbfd.a
-libtool: finish: PATH="/home/thomas/command-i686:/home/thomas/command:/home/thomas/shared/command:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/games:/sbin" ldconfig -n [...]/tschwinge/Ferry_Tagscherer.build.install/lib
 ----------------------------------------------------------------------
 Libraries have been installed in:
    [...]/tschwinge/Ferry_Tagscherer.build.install/lib
@@ -104,7 +103,6 @@
 libtool: install: /usr/bin/install -c .libs/libopcodes.a [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libopcodes.a
 libtool: install: chmod 644 [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libopcodes.a
 libtool: install: ranlib [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libopcodes.a
-libtool: finish: PATH="/home/thomas/command-i686:/home/thomas/command:/home/thomas/shared/command:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/games:/sbin" ldconfig -n [...]/tschwinge/Ferry_Tagscherer.build.install/lib
 ----------------------------------------------------------------------
 Libraries have been installed in:
    [...]/tschwinge/Ferry_Tagscherer.build.install/lib
@@ -241,8 +239,6 @@
 make[4]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/doc'
 make[4]: Nothing to be done for `all'.
 make[4]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/doc'
-make[4]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[4]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
 make[4]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/data-directory'
 make[4]: Nothing to be done for `all'.
 make[4]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/data-directory'
@@ -337,20 +333,6 @@
 make[5]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/testsuite'
 make[5]: Nothing to be done for `install'.
 make[5]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/testsuite'
-make[5]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[6]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-n=`echo gdbserver | sed 's,y,y,'`; \
-	if [ x$n = x ]; then n=gdbserver; else true; fi; \
-	if [ x"i386-linux-ipa.o linux-i386-ipa.o" != x ]; then \
-		/bin/sh ../../../Ferry_Tagscherer/gdb/gdbserver/../../mkinstalldirs [...]/tschwinge/Ferry_Tagscherer.build.install/lib; \
-		/usr/bin/install -c libinproctrace.so [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libinproctrace.so; \
-	fi; \
-	/bin/sh ../../../Ferry_Tagscherer/gdb/gdbserver/../../mkinstalldirs [...]/tschwinge/Ferry_Tagscherer.build.install/bin; \
-	/usr/bin/install -c gdbserver [...]/tschwinge/Ferry_Tagscherer.build.install/bin/$n; \
-	/bin/sh ../../../Ferry_Tagscherer/gdb/gdbserver/../../mkinstalldirs [...]/tschwinge/Ferry_Tagscherer.build.install/share/man/man1; \
-	/usr/bin/install -c -m 644 ../../../Ferry_Tagscherer/gdb/gdbserver/gdbserver.1 [...]/tschwinge/Ferry_Tagscherer.build.install/share/man/man1/$n.1
-make[6]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[5]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
 make[5]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/data-directory'
 make[6]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/data-directory'
 /bin/dash ../../../Ferry_Tagscherer/gdb/data-directory/../../mkinstalldirs [...]/tschwinge/Ferry_Tagscherer.build.install/share/gdb/syscalls
