--- toolchain/logs/gdb/kepler.SCHWINGE/install_	2012-11-11 00:46:41.247624041 +0100
+++ toolchain/logs/gdb/coulomb.SCHWINGE/install_	2012-11-11 00:46:41.255624043 +0100
@@ -70,7 +70,6 @@
 libtool: install: /usr/bin/install -c .libs/libbfd.a [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libbfd.a
 libtool: install: chmod 644 [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libbfd.a
 libtool: install: ranlib [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libbfd.a
-libtool: finish: PATH="/home/thomas/command-i686:/home/thomas/command:/home/thomas/shared/command:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/games:/sbin" ldconfig -n [...]/tschwinge/Ferry_Tagscherer.build.install/lib
 ----------------------------------------------------------------------
 Libraries have been installed in:
    [...]/tschwinge/Ferry_Tagscherer.build.install/lib
@@ -96,19 +95,16 @@
 make[2]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/opcodes'
 Making install in .
 make[3]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/opcodes'
-../../Ferry_Tagscherer/opcodes/i386-tbl.h
 make[4]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/opcodes'
 make[4]: Nothing to be done for `install-exec-am'.
 test -z "[...]/tschwinge/Ferry_Tagscherer.build.install/include" || /bin/mkdir -p "[...]/tschwinge/Ferry_Tagscherer.build.install/include"
  /usr/bin/install -c -m 644 ../../Ferry_Tagscherer/opcodes/../include/dis-asm.h '[...]/tschwinge/Ferry_Tagscherer.build.install/include'
-../../Ferry_Tagscherer/opcodes/i386-tbl.h
 test -z "[...]/tschwinge/Ferry_Tagscherer.build.install/lib" || /bin/mkdir -p "[...]/tschwinge/Ferry_Tagscherer.build.install/lib"
  /bin/dash ./libtool   --mode=install /usr/bin/install -c   libopcodes.la '[...]/tschwinge/Ferry_Tagscherer.build.install/lib'
 libtool: install: /usr/bin/install -c .libs/libopcodes.lai [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libopcodes.la
 libtool: install: /usr/bin/install -c .libs/libopcodes.a [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libopcodes.a
 libtool: install: chmod 644 [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libopcodes.a
 libtool: install: ranlib [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libopcodes.a
-libtool: finish: PATH="/home/thomas/command-i686:/home/thomas/command:/home/thomas/shared/command:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/games:/sbin" ldconfig -n [...]/tschwinge/Ferry_Tagscherer.build.install/lib
 ----------------------------------------------------------------------
 Libraries have been installed in:
    [...]/tschwinge/Ferry_Tagscherer.build.install/lib
@@ -251,36 +247,6 @@
 make[4]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/doc'
 make[4]: Nothing to be done for `all'.
 make[4]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/doc'
-make[4]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[5]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[6]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[7]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[8]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make  all-recursive
-make[9]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[10]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[10]: Nothing to be done for `all-am'.
-make[10]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[9]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[8]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[7]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[6]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[5]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[5]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[6]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[7]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[8]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make  all-recursive
-make[9]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[10]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[10]: Nothing to be done for `all-am'.
-make[10]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[9]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[8]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[7]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[6]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[5]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[4]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
 make[4]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/data-directory'
 make[4]: Nothing to be done for `all'.
 make[4]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/data-directory'
@@ -376,90 +342,6 @@
 make[5]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/testsuite'
 make[5]: Nothing to be done for `install'.
 make[5]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/testsuite'
-make[5]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[6]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[7]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[8]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[9]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make  all-recursive
-make[10]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[11]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[11]: Nothing to be done for `all-am'.
-make[11]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[10]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[9]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[8]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[7]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[6]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[6]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[7]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[8]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[9]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make  all-recursive
-make[10]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[11]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[11]: Nothing to be done for `all-am'.
-make[11]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[10]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[9]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[8]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[7]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[6]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[6]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-n=`echo gdbserver | sed 's,y,y,'`; \
-	if [ x$n = x ]; then n=gdbserver; else true; fi; \
-	if [ x"i386-linux-ipa.o linux-i386-ipa.o" != x ]; then \
-		/bin/sh ../../../Ferry_Tagscherer/gdb/gdbserver/../../mkinstalldirs [...]/tschwinge/Ferry_Tagscherer.build.install/lib; \
-		/usr/bin/install -c libinproctrace.so [...]/tschwinge/Ferry_Tagscherer.build.install/lib/libinproctrace.so; \
-	fi; \
-	/bin/sh ../../../Ferry_Tagscherer/gdb/gdbserver/../../mkinstalldirs [...]/tschwinge/Ferry_Tagscherer.build.install/bin; \
-	/usr/bin/install -c gdbserver [...]/tschwinge/Ferry_Tagscherer.build.install/bin/$n; \
-	/bin/sh ../../../Ferry_Tagscherer/gdb/gdbserver/../../mkinstalldirs [...]/tschwinge/Ferry_Tagscherer.build.install/share/man/man1; \
-	/usr/bin/install -c -m 644 ../../../Ferry_Tagscherer/gdb/gdbserver/gdbserver.1 [...]/tschwinge/Ferry_Tagscherer.build.install/share/man/man1/$n.1
-make[7]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[8]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[9]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[10]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make  install-recursive
-make[11]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[12]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[13]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-if test yes = no; then \
-	  case 'linux-gnu' in \
-	    darwin[56]*) \
-	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
-	      need_charset_alias=false ;; \
-	    *) \
-	      need_charset_alias=true ;; \
-	  esac ; \
-	else \
-	  need_charset_alias=false ; \
-	fi ; \
-	if $need_charset_alias; then \
-	  /bin/dash ../../../../../Ferry_Tagscherer/gdb/gdbserver/../gnulib/../../mkinstalldirs [...]/tschwinge/Ferry_Tagscherer.build.install/lib ; \
-	fi ; \
-	if test -f [...]/tschwinge/Ferry_Tagscherer.build.install/lib/charset.alias; then \
-	  sed -f ref-add.sed [...]/tschwinge/Ferry_Tagscherer.build.install/lib/charset.alias > [...]/tschwinge/Ferry_Tagscherer.build.install/lib/charset.tmp ; \
-	  /usr/bin/install -c -m 644 [...]/tschwinge/Ferry_Tagscherer.build.install/lib/charset.tmp [...]/tschwinge/Ferry_Tagscherer.build.install/lib/charset.alias ; \
-	  rm -f [...]/tschwinge/Ferry_Tagscherer.build.install/lib/charset.tmp ; \
-	else \
-	  if $need_charset_alias; then \
-	    sed -f ref-add.sed charset.alias > [...]/tschwinge/Ferry_Tagscherer.build.install/lib/charset.tmp ; \
-	    /usr/bin/install -c -m 644 [...]/tschwinge/Ferry_Tagscherer.build.install/lib/charset.tmp [...]/tschwinge/Ferry_Tagscherer.build.install/lib/charset.alias ; \
-	    rm -f [...]/tschwinge/Ferry_Tagscherer.build.install/lib/charset.tmp ; \
-	  fi ; \
-	fi
-make[13]: Nothing to be done for `install-data-am'.
-make[13]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[12]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[11]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[10]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver/import'
-make[9]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[8]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver/build-gnulib-gdbserver'
-make[7]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[6]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
-make[5]: Leaving directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/gdbserver'
 make[5]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/data-directory'
 make[6]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/data-directory'
 /bin/dash ../../../Ferry_Tagscherer/gdb/data-directory/../../mkinstalldirs [...]/tschwinge/Ferry_Tagscherer.build.install/share/gdb/syscalls
@@ -500,7 +382,7 @@
 make[10]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/build-gnulib/import'
 make[11]: Entering directory `[...]/tschwinge/Ferry_Tagscherer.build/gdb/build-gnulib/import'
 if test yes = no; then \
-	  case 'linux-gnu' in \
+	  case 'gnu0.3' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
 	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
