--- /dev/fd/63	2011-09-06 18:18:41.081646850 +0200
+++ /dev/fd/62	2011-09-06 18:18:41.081646850 +0200
@@ -68,7 +68,6 @@
 libtool: install: /usr/bin/install -c .libs/libbfd.a [...]/hurd/master.build.install/lib/libbfd.a
 libtool: install: chmod 644 [...]/hurd/master.build.install/lib/libbfd.a
 libtool: install: ranlib [...]/hurd/master.build.install/lib/libbfd.a
-libtool: finish: PATH="/home/thomas/command-i686:/home/thomas/command:/home/thomas/shared/command:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/games:/sbin" ldconfig -n [...]/hurd/master.build.install/lib
 ----------------------------------------------------------------------
 Libraries have been installed in:
    [...]/hurd/master.build.install/lib
@@ -104,7 +103,6 @@
 libtool: install: /usr/bin/install -c .libs/libopcodes.a [...]/hurd/master.build.install/lib/libopcodes.a
 libtool: install: chmod 644 [...]/hurd/master.build.install/lib/libopcodes.a
 libtool: install: ranlib [...]/hurd/master.build.install/lib/libopcodes.a
-libtool: finish: PATH="/home/thomas/command-i686:/home/thomas/command:/home/thomas/shared/command:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/games:/sbin" ldconfig -n [...]/hurd/master.build.install/lib
 ----------------------------------------------------------------------
 Libraries have been installed in:
    [...]/hurd/master.build.install/lib
@@ -241,8 +239,6 @@
 make[4]: Entering directory `[...]/hurd/master.build/gdb/doc'
 make[4]: Nothing to be done for `all'.
 make[4]: Leaving directory `[...]/hurd/master.build/gdb/doc'
-make[4]: Entering directory `[...]/hurd/master.build/gdb/gdbserver'
-make[4]: Leaving directory `[...]/hurd/master.build/gdb/gdbserver'
 make[4]: Entering directory `[...]/hurd/master.build/gdb/data-directory'
 make[4]: Nothing to be done for `all'.
 make[4]: Leaving directory `[...]/hurd/master.build/gdb/data-directory'
@@ -337,20 +333,6 @@
 make[5]: Entering directory `[...]/hurd/master.build/gdb/testsuite'
 make[5]: Nothing to be done for `install'.
 make[5]: Leaving directory `[...]/hurd/master.build/gdb/testsuite'
-make[5]: Entering directory `[...]/hurd/master.build/gdb/gdbserver'
-make[6]: Entering directory `[...]/hurd/master.build/gdb/gdbserver'
-n=`echo gdbserver | sed 's,y,y,'`; \
-	if [ x$n = x ]; then n=gdbserver; else true; fi; \
-	if [ x"i386-linux-ipa.o linux-i386-ipa.o" != x ]; then \
-		/bin/sh ../../../master/gdb/gdbserver/../../mkinstalldirs [...]/hurd/master.build.install/lib; \
-		/usr/bin/install -c libinproctrace.so [...]/hurd/master.build.install/lib/libinproctrace.so; \
-	fi; \
-	/bin/sh ../../../master/gdb/gdbserver/../../mkinstalldirs [...]/hurd/master.build.install/bin; \
-	/usr/bin/install -c gdbserver [...]/hurd/master.build.install/bin/$n; \
-	/bin/sh ../../../master/gdb/gdbserver/../../mkinstalldirs [...]/hurd/master.build.install/share/man/man1; \
-	/usr/bin/install -c -m 644 ../../../master/gdb/gdbserver/gdbserver.1 [...]/hurd/master.build.install/share/man/man1/$n.1
-make[6]: Leaving directory `[...]/hurd/master.build/gdb/gdbserver'
-make[5]: Leaving directory `[...]/hurd/master.build/gdb/gdbserver'
 make[5]: Entering directory `[...]/hurd/master.build/gdb/data-directory'
 make[6]: Entering directory `[...]/hurd/master.build/gdb/data-directory'
 /bin/dash ../../../master/gdb/data-directory/../../mkinstalldirs [...]/hurd/master.build.install/share/gdb/syscalls
