--- /dev/fd/63	2012-06-08 15:56:29.038140510 +0200
+++ /dev/fd/62	2012-06-08 15:56:29.038140510 +0200
@@ -52,588 +52,192 @@
 make  check-TESTS
 make[2]: Entering directory `[...]/master.build'
 Completed 6 tests
-Allocated 10285328 collectable objects
+Allocated 10285344 collectable objects
 Allocated 1212 uncollectable objects
 Allocated 7500000 atomic objects
-Allocated 137745 stubborn objects
-Finalized 6625/6625 objects - finalization is probably ok
-Total number of bytes allocated is 1148245774
-Final heap size is 28180480 bytes
+Allocated 137760 stubborn objects
+Finalized 6610/6610 objects - finalization is probably ok
+Total number of bytes allocated is 794703657
+Final heap size is 20897792 bytes
 Collector appears to work
-Completed 277 collections
+Completed 161 collections
 PASS: gctest
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 PASS: leaktest
-Final heap size is 131072
+Final heap size is 1536000
 PASS: middletest
 GC_check_heap_block: found smashed heap objects:
 [ADDR] in or near object at [ADDR](../master/tests/smash_test.c:21, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:21 [[ADDR]]
 GC_check_heap_block: found smashed heap objects:
 [ADDR] in or near object at [ADDR](../master/tests/smash_test.c:21, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:21 [[ADDR]]
-GC_check_heap_block: found smashed heap objects:
-[ADDR] in or near object at [ADDR](../master/tests/smash_test.c:21, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:21 [[ADDR]]
 [ADDR] in or near object at [ADDR](../master/tests/smash_test.c:21, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:21 [[ADDR]]
 PASS: smashtest
 PASS: hugetest
 PASS: staticrootstest
 
 ****Chose address [ADDR] in object
-[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
+[ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
 
-Reachable via 0 levels of pointers from offset 0 in object:
+Reachable via 0 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 1 levels of pointers from offset 0 in object:
+Reachable via 1 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 2 levels of pointers from offset 4 in object:
+Reachable via 2 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 3 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 4 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 5 levels of pointers from offset 4 in object:
+Reachable via 5 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 6 levels of pointers from offset 0 in object:
+Reachable via 6 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 7 levels of pointers from offset 0 in object:
+Reachable via 7 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 8 levels of pointers from offset 0 in object:
+Reachable via 8 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
-
-Reachable via 9 levels of pointers from offset 0 in object:
-[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 10 levels of pointers from root at [ADDR]
+Reachable via 9 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
+
+Reachable via 10 levels of pointers from offset 0 in object:
+[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
+
+Reachable via 11 levels of pointers from root at [ADDR]
 
 
 ****Chose address [ADDR] in object
-[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
+[ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
 
 Reachable via 0 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 1 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 2 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 3 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 4 levels of pointers from offset 4 in object:
+Reachable via 4 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 5 levels of pointers from offset 0 in object:
+Reachable via 5 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 6 levels of pointers from offset 4 in object:
+Reachable via 6 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 7 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 8 levels of pointers from offset 4 in object:
+Reachable via 8 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 9 levels of pointers from root at [ADDR]
+Reachable via 9 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
+
+Reachable via 10 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
+
+Reachable via 11 levels of pointers from root at [ADDR]
 
 
 ****Chose address [ADDR] in object
-[ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc_atomic+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
+[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
 
-Reachable via 0 levels of pointers from offset 0 in object:
+Reachable via 0 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 1 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 2 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 3 levels of pointers from offset 4 in object:
+Reachable via 3 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 4 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 5 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 6 levels of pointers from offset 0 in object:
+Reachable via 6 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 7 levels of pointers from offset 0 in object:
+Reachable via 7 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 8 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 9 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 10 levels of pointers from offset 4 in object:
-[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
-
-Reachable via 11 levels of pointers from root at [ADDR]
+Reachable via 10 levels of pointers from root at [ADDR]
 
 
 ****Chose address [ADDR] in object
 [ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc_atomic+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
-Reachable via 0 levels of pointers from offset 4 in object:
+Reachable via 0 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 1 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 2 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
-Reachable via 3 levels of pointers from offset 0 in object:
+Reachable via 3 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 4 levels of pointers from offset 4 in object:
+Reachable via 4 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 5 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 6 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 7 levels of pointers from offset 4 in object:
+Reachable via 7 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 8 levels of pointers from offset 4 in object:
+Reachable via 8 levels of pointers from offset 0 in object:
+[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
+
+Reachable via 9 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
-
-Reachable via 9 levels of pointers from offset 0 in object:
-[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 10 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 11 levels of pointers from root at [ADDR]
 
 PASS: tracetest
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:12, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:12 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:12, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:12 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:12, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:12 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:12, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:12 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:12, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:12 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 PASS: threadleaktest
 usage: test_cpp number-of-iterations
