diff --git toolchain/logs/boehm-gc/kepler.SCHWINGE/log_test_ toolchain/logs/boehm-gc/laplace.SCHWINGE/log_test_
index e323bc7..4a6e695 100644
--- toolchain/logs/boehm-gc/kepler.SCHWINGE/log_test_
+++ toolchain/logs/boehm-gc/laplace.SCHWINGE/log_test_
@@ -82,85 +82,48 @@ make[2]: Entering directory '[...]/master.build'
 SUCCEEDED
 PASS: cordtest
 Completed 6 tests
-Allocated 10285707 collectable objects
+Allocated 10285267 collectable objects
 Allocated 1212 uncollectable objects
-Allocated 7499997 atomic objects
-Allocated 137759 stubborn objects
-Finalized 6621/6621 objects - finalization is probably ok
-Total number of bytes allocated is 1147021510
-Final heap size is 29728768 bytes
-Completed 277 collections (using 2 marker threads)
+Allocated 7500000 atomic objects
+Allocated 137760 stubborn objects
+Finalized 6617/6617 objects - finalization is probably ok
+Total number of bytes allocated is 795596801
+Final heap size is 21819392 bytes
+Completed 209 collections
 Collector appears to work
 PASS: gctest
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 (discriminator 2) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=5, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 (discriminator 2) [[ADDR]]
+
+Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=6, NORMAL)
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=7, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 (discriminator 2) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 (discriminator 2) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=9, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 (discriminator 2) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=10, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 (discriminator 2) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=11, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 (discriminator 2) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=12, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 (discriminator 2) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:12, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:12 (discriminator 2) [[ADDR]]
 
 PASS: leaktest
 Final heap size is 131072
 PASS: middletest
 GC_check_heap_block: found smashed heap objects:
  [ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
 
 GC_check_heap_block: found smashed heap objects:
  [ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
 
 GC_check_heap_block: found smashed heap objects:
  [ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
  [ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
 
 PASS: smashtest
 PASS: hugetest
@@ -170,550 +133,171 @@ PASS: realloc_test
 PASS: staticrootstest
 
 ****Chosen address [ADDR] in object
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
+[ADDR] (../master/tests/trace_test.c:19, sz=8, PTRFREE)
 
 Reachable via 0 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
 
 Reachable via 1 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
 
-Reachable via 2 levels of pointers from offset 0 in object:
+Reachable via 2 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
 
 Reachable via 3 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
-Reachable via 4 levels of pointers from offset 4 in object:
+Reachable via 4 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 5 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 6 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 7 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 8 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
-Reachable via 9 levels of pointers from root at [ADDR]
+Reachable via 9 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 10 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 11 levels of pointers from root at [ADDR]
 
 
 ****Chosen address [ADDR] in object
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
+[ADDR] (../master/tests/trace_test.c:19, sz=8, PTRFREE)
 
 Reachable via 0 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
 
-Reachable via 1 levels of pointers from offset 4 in object:
+Reachable via 1 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
 
 Reachable via 2 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
 
 Reachable via 3 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
 
-Reachable via 4 levels of pointers from offset 4 in object:
+Reachable via 4 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
-Reachable via 5 levels of pointers from offset 4 in object:
+Reachable via 5 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 6 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
-Reachable via 7 levels of pointers from offset 0 in object:
+Reachable via 7 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 8 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 9 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
+
+Reachable via 10 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 11 levels of pointers from root at [ADDR]
+
+
+****Chosen address [ADDR] in object
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 0 levels of pointers from offset 0 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 1 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 2 levels of pointers from offset 0 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 3 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 4 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 5 levels of pointers from offset 0 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 6 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 7 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 8 levels of pointers from offset 0 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
+
+Reachable via 9 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
 
 Reachable via 10 levels of pointers from root at [ADDR]
 
 
 ****Chosen address [ADDR] in object
 [ADDR] (../master/tests/trace_test.c:19, sz=8, PTRFREE)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc_atomic+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 (discriminator 1) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
 
-Reachable via 0 levels of pointers from offset 0 in object:
+Reachable via 0 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
 
-Reachable via 1 levels of pointers from offset 0 in object:
+Reachable via 1 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
 
 Reachable via 2 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
 
 Reachable via 3 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
 
 Reachable via 4 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
 
 Reachable via 5 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 6 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 7 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 8 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 9 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 10 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
-
-Reachable via 11 levels of pointers from root at [ADDR]
-
-
-****Chosen address [ADDR] in object
-[ADDR] (../master/tests/trace_test.c:19, sz=8, PTRFREE)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc_atomic+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 (discriminator 1) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-
-Reachable via 0 levels of pointers from offset 4 in object:
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-
-Reachable via 1 levels of pointers from offset 4 in object:
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-
-Reachable via 2 levels of pointers from offset 0 in object:
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-
-Reachable via 3 levels of pointers from offset 0 in object:
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-
-Reachable via 4 levels of pointers from offset 4 in object:
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-
-Reachable via 5 levels of pointers from offset 4 in object:
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
-
-Reachable via 6 levels of pointers from offset 0 in object:
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
-
-Reachable via 7 levels of pointers from offset 4 in object:
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:25 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
-
-Reachable via 8 levels of pointers from offset 4 in object:
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
-
-Reachable via 9 levels of pointers from offset 0 in object:
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:24 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
-
-Reachable via 10 levels of pointers from offset 0 in object:
-[ADDR] (../master/tests/trace_test.c:17, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:34 (discriminator 2) [[ADDR]]
 
 Reachable via 11 levels of pointers from root at [ADDR]
 
 PASS: tracetest
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:25, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:25 (discriminator 2) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:25, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:25 (discriminator 2) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:25, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:25 (discriminator 2) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:25, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:25 (discriminator 2) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:25, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:25 (discriminator 2) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 PASS: threadleaktest
 PASS: threadkey_test
-subthread_create: created 184 threads (180 ended)
+subthread_create: created 201 threads (201 ended)
 PASS: subthread_create
 PASS: initsecondarythread
 usage: test_cpp number-of-iterations
@@ -733,9 +317,9 @@ PASS: test_cpp
 Threaded disclaim test.
 PASS: disclaim_test
 			fin. ratio       time/s    time/fin.
-regular finalization:       0.9922       35.842  8.61269e-06
- finalize on reclaim:       0.9998        0.487  1.16137e-07
-     no finalization:       0.0000        0.405          N/A
+regular finalization:       0.9922            0            0
+ finalize on reclaim:       0.9996            0            0
+     no finalization:       0.0000            0          N/A
 PASS: disclaim_bench
 ===================
 All 16 tests passed
@@ -887,10 +471,10 @@ Succeeded
 Testing test_and_set
 Succeeded
 PASS: test_atomic_pthreads
-About 1000000 pushes + 1000000 pops in 1 threads: 34 msecs
-About 1000000 pushes + 1000000 pops in 2 threads: 67 msecs
-About 1000000 pushes + 1000000 pops in 3 threads: 113 msecs
-About 1000000 pushes + 1000000 pops in 4 threads: 93 msecs
+About 1000000 pushes + 1000000 pops in 1 threads: 20 msecs
+About 1000000 pushes + 1000000 pops in 2 threads: 20 msecs
+About 1000000 pushes + 1000000 pops in 3 threads: 30 msecs
+About 1000000 pushes + 1000000 pops in 4 threads: 30 msecs
 PASS: test_stack
 Testing AO_malloc/AO_free
 Succeeded
