diff --git toolchain/logs/boehm-gc/kepler.SCHWINGE/log_build_ toolchain/logs/boehm-gc/laplace.SCHWINGE/log_build_
index 7d906b4..13216b0 100644
--- toolchain/logs/boehm-gc/kepler.SCHWINGE/log_build_
+++ toolchain/logs/boehm-gc/laplace.SCHWINGE/log_build_
@@ -35,7 +35,7 @@ checking if the linker (/usr/bin/ld) is GNU ld... yes
 checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
 checking the name lister (/usr/bin/nm -B) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 1572864
+checking the maximum length of command line arguments... -1
 checking how to convert [ARCH] file names to [ARCH] format... func_convert_file_noop
 checking how to convert [ARCH] file names to toolchain format... func_convert_file_noop
 checking for /usr/bin/ld option to reload object files... -r
@@ -95,7 +95,6 @@ checking how to hardcode library paths into programs... immediate
 checking for thread model used by GCC... posix
 checking for inline... inline
 checking for pthread_self in -lpthread... yes
-configure: WARNING: "Explicit GC_INIT() calls may be required."
 checking for gcc -Wextra... yes
 checking for xlc... no
 checking whether gcc supports -fno-strict-aliasing... yes
@@ -104,7 +103,6 @@ checking whether to build shared libraries... yes
 checking which machine-dependent code should be used... 
 checking whether Solaris gcc optimization fix is necessary... no
 configure: WARNING: "Should define GC_DEBUG and use debug alloc in clients."
-configure: WARNING: "Client must not use -fomit-frame-pointer."
 checking for dladdr... yes
 checking sys/dg_sys_info.h usability... no
 checking sys/dg_sys_info.h presence... no
@@ -156,7 +154,7 @@ checking if the linker (/usr/bin/ld) is GNU ld... yes
 checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
 checking the name lister (/usr/bin/nm -B) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 1572864
+checking the maximum length of command line arguments... -1
 checking how to convert [ARCH] file names to [ARCH] format... func_convert_file_noop
 checking how to convert [ARCH] file names to toolchain format... func_convert_file_noop
 checking for /usr/bin/ld option to reload object files... -r
@@ -401,7 +399,7 @@ mv -f $depbase.Tpo $depbase.Plo
 libtool: compile:  g++-4.9 -DHAVE_CONFIG_H -I./include -I../master/include -I./libatomic_ops/src -I../master/libatomic_ops/src -fexceptions -g -O2 -MT gc_cpp.lo -MD -MP -MF .deps/gc_cpp.Tpo -c ../master/gc_cpp.cc  -fPIC -DPIC -o .libs/gc_cpp.o
 libtool: compile:  g++-4.9 -DHAVE_CONFIG_H -I./include -I../master/include -I./libatomic_ops/src -I../master/libatomic_ops/src -fexceptions -g -O2 -MT gc_cpp.lo -MD -MP -MF .deps/gc_cpp.Tpo -c ../master/gc_cpp.cc -o gc_cpp.o >/dev/null 2>&1
 /bin/bash ./libtool  --tag=CXX   --mode=link g++-4.9 -fexceptions -g -O2 -version-info 1:3:0 -no-undefined  -o libgccpp.la -rpath [...]/master.build.install/lib gc_cpp.lo ./libgc.la 
-libtool: link: g++-4.9  -fPIC -DPIC -shared -nostdlib /usr/lib/gcc/i586-linux-gnu/4.9/../../../i386-linux-gnu/crti.o /usr/lib/gcc/i586-linux-gnu/4.9/crtbeginS.o  .libs/gc_cpp.o   -Wl,-rpath -Wl,[...]/master.build/.libs -Wl,-rpath -Wl,[...]/master.build.install/lib ./.libs/libgc.so -L/usr/lib/gcc/i586-linux-gnu/4.9 -L/usr/lib/gcc/i586-linux-gnu/4.9/../../../i386-linux-gnu -L/usr/lib/gcc/i586-linux-gnu/4.9/../../../../lib -L/lib/i386-linux-gnu -L/lib/../lib -L/usr/lib/i386-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/i586-linux-gnu/4.9/../../.. -lstdc++ -lm -lc -lgcc_s /usr/lib/gcc/i586-linux-gnu/4.9/crtendS.o /usr/lib/gcc/i586-linux-gnu/4.9/../../../i386-linux-gnu/crtn.o  -g -O2   -Wl,-soname -Wl,libgccpp.so.1 -o .libs/libgccpp.so.1.0.3
+libtool: link: g++-4.9  -fPIC -DPIC -shared -nostdlib /usr/lib/gcc/i586-gnu/4.9/../../../i386-gnu/crti.o /usr/lib/gcc/i586-gnu/4.9/crtbeginS.o  .libs/gc_cpp.o   -Wl,-rpath -Wl,[...]/master.build/.libs -Wl,-rpath -Wl,[...]/master.build.install/lib ./.libs/libgc.so -L/usr/lib/gcc/i586-gnu/4.9 -L/usr/lib/gcc/i586-gnu/4.9/../../../i386-gnu -L/usr/lib/gcc/i586-gnu/4.9/../../.. -L/lib/i386-gnu -L/usr/lib/i386-gnu -lstdc++ -lm -lc -lgcc_s /usr/lib/gcc/i586-gnu/4.9/crtendS.o /usr/lib/gcc/i586-gnu/4.9/../../../i386-gnu/crtn.o  -g -O2   -Wl,-soname -Wl,libgccpp.so.1 -o .libs/libgccpp.so.1.0.3
 libtool: link: (cd ".libs" && rm -f "libgccpp.so.1" && ln -s "libgccpp.so.1.0.3" "libgccpp.so.1")
 libtool: link: (cd ".libs" && rm -f "libgccpp.so" && ln -s "libgccpp.so.1.0.3" "libgccpp.so")
 libtool: link: ar cru .libs/libgccpp.a  gc_cpp.o
