--- /dev/fd/63	2012-06-09 09:23:26.967838989 +0200
+++ /dev/fd/62	2012-06-09 09:23:26.967838989 +0200
@@ -64,646 +64,223 @@
 make  check-TESTS
 make[2]: Entering directory `[...]/master.build'
 Completed 6 tests
-Allocated 10285404 collectable objects
+Allocated 10285258 collectable objects
 Allocated 1212 uncollectable objects
 Allocated 7500000 atomic objects
-Allocated 137750 stubborn objects
-Finalized 6616/6616 objects - finalization is probably ok
-Total number of bytes allocated is 1136258462
-Final heap size is 27959296 bytes
+Allocated 137760 stubborn objects
+Finalized 6609/6609 objects - finalization is probably ok
+Total number of bytes allocated is 796643017
+Final heap size is 27873280 bytes
 Collector appears to work
-Completed 275 collections
+Completed 126 collections
 PASS: gctest
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=5, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=6, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=7, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=9, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=10, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=11, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=12, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 PASS: leaktest
-Final heap size is 131072
-PASS: middletest
+Assertion failure: ../master/thread_local_alloc.c:156
+assertion failure
+/bin/bash: line 5: 24743 Aborted                 ${dir}$tst
+FAIL: middletest
 GC_check_heap_block: found smashed heap objects:
  [ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
 
 GC_check_heap_block: found smashed heap objects:
  [ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
-
-GC_check_heap_block: found smashed heap objects:
  [ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
- [ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
 
 PASS: smashtest
 PASS: hugetest
 Heap size: 65536
 Heap size: 131072
+Heap size: 196608
+Heap size: 266240
+Heap size: 356352
+Heap size: 479232
+Heap size: 643072
+Heap size: 860160
+Heap size: 1150976
+Heap size: 1536000
 PASS: realloc_test
 PASS: staticrootstest
 
 ****Chosen address [ADDR] in object
-[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
+[ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
 
-Reachable via 0 levels of pointers from offset 0 in object:
+Reachable via 0 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 1 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 2 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 3 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 4 levels of pointers from offset 4 in object:
+Reachable via 4 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 5 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 6 levels of pointers from offset 0 in object:
+Reachable via 6 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 7 levels of pointers from offset 0 in object:
+Reachable via 7 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 8 levels of pointers from offset 0 in object:
+Reachable via 8 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 9 levels of pointers from root at [ADDR]
+Reachable via 9 levels of pointers from offset 4 in object:
+[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
+
+Reachable via 10 levels of pointers from offset 0 in object:
+[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
+
+Reachable via 11 levels of pointers from root at [ADDR]
 
 
 ****Chosen address [ADDR] in object
 [ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc_atomic+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 0 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 1 levels of pointers from offset 0 in object:
+Reachable via 1 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 2 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 3 levels of pointers from offset 4 in object:
+Reachable via 3 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
-Reachable via 4 levels of pointers from offset 4 in object:
+Reachable via 4 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 5 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 6 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 7 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 8 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 9 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 10 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 11 levels of pointers from root at [ADDR]
 
 
 ****Chosen address [ADDR] in object
-[ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc_atomic+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
+[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
 
 Reachable via 0 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 1 levels of pointers from offset 4 in object:
+Reachable via 1 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
-Reachable via 2 levels of pointers from offset 0 in object:
+Reachable via 2 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 3 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 4 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 5 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 6 levels of pointers from offset 0 in object:
+Reachable via 6 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 7 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 8 levels of pointers from offset 4 in object:
+Reachable via 8 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
-
-Reachable via 9 levels of pointers from offset 0 in object:
-[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 10 levels of pointers from offset 0 in object:
+Reachable via 9 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 11 levels of pointers from root at [ADDR]
+Reachable via 10 levels of pointers from root at [ADDR]
 
 
 ****Chosen address [ADDR] in object
 [ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc_atomic+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 0 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
-Reachable via 1 levels of pointers from offset 0 in object:
+Reachable via 1 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 2 levels of pointers from offset 4 in object:
+Reachable via 2 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 3 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
-Reachable via 4 levels of pointers from offset 4 in object:
+Reachable via 4 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 5 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 6 levels of pointers from offset 4 in object:
+Reachable via 6 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 7 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 8 levels of pointers from offset 4 in object:
+Reachable via 8 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 9 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 10 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 11 levels of pointers from root at [ADDR]
 
 PASS: tracetest
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:25, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:25 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:25, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:25 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:25, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:25 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:25, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:25 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:25, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:25 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 PASS: threadleaktest
 PASS: threadkey_test
@@ -722,11 +299,16 @@
 Starting iteration 10
 The test appears to have succeeded.
 PASS: test_cpp
-===================
-All 12 tests passed
-===================
+==================================
+1 of 12 tests failed
+Please report to hans.boehm@hp.com
+==================================
+make[2]: *** [check-TESTS] Error 1
 make[2]: Leaving directory `[...]/master.build'
+make[1]: *** [check-am] Error 2
 make[1]: Leaving directory `[...]/master.build'
+make: *** [check-recursive] Error 1
+make: Target `check' not remade because of errors.
 Making check in src
 make[1]: Entering directory `[...]/master.build/libatomic_ops/src'
 /bin/bash ../libtool --tag=CC   --mode=compile gcc-4.6 -DHAVE_CONFIG_H   -I../src -I../../../master/libatomic_ops/src  -fPIC -g -O2 -Wall -Wextra -DNDEBUG -MT atomic_ops.lo -MD -MP -MF .deps/atomic_ops.Tpo -c -o atomic_ops.lo ../../../master/libatomic_ops/src/atomic_ops.c
@@ -872,24 +454,21 @@
 Testing test_and_set
 Succeeded
 PASS: test_atomic_pthreads
-FAILED
-/bin/bash: line 5: 27329 Aborted                 ${dir}$tst
-FAIL: test_stack
+About 1000000 pushes + 1000000 pops in 1 threads: 80 msecs
+About 1000000 pushes + 1000000 pops in 2 threads: 70 msecs
+About 1000000 pushes + 1000000 pops in 3 threads: 90 msecs
+About 1000000 pushes + 1000000 pops in 4 threads: 70 msecs
+PASS: test_stack
 Testing AO_malloc/AO_free
 Succeeded
 Performing 1000 reversals of 1000 element lists in 10 threads
 PASS: test_malloc
-===================
-1 of 4 tests failed
-===================
-make[3]: *** [check-TESTS] Error 1
+==================
+All 4 tests passed
+==================
 make[3]: Leaving directory `[...]/master.build/libatomic_ops/tests'
-make[2]: *** [check-am] Error 2
 make[2]: Leaving directory `[...]/master.build/libatomic_ops/tests'
-make[1]: *** [check] Error 2
 make[1]: Leaving directory `[...]/master.build/libatomic_ops/tests'
 make[1]: Entering directory `[...]/master.build/libatomic_ops'
 make[1]: Nothing to be done for `check-am'.
 make[1]: Leaving directory `[...]/master.build/libatomic_ops'
-make: *** [check-recursive] Error 1
-make: Target `check' not remade because of errors.
