diff --git toolchain/logs/boehm-gc/kepler.SCHWINGE/log_install_ toolchain/logs/boehm-gc/laplace.SCHWINGE/log_install_
index 3e74daf..dc9b905 100644
--- toolchain/logs/boehm-gc/kepler.SCHWINGE/log_install_
+++ toolchain/logs/boehm-gc/laplace.SCHWINGE/log_install_
@@ -8,7 +8,7 @@ libtool: install: (cd [...]/master.build.install/lib && { ln -s -f libgc.so.1.0.
 libtool: install: /usr/bin/install -c .libs/libgc.lai [...]/master.build.install/lib/libgc.la
 libtool: warning: relinking 'libgccpp.la'
 libtool: install: (cd [...]/master.build; /bin/bash "[...]/master.build/libtool"  --tag CXX --mode=relink g++-4.9 -fexceptions -g -O2 -version-info 1:3:0 -no-undefined -o libgccpp.la -rpath [...]/master.build.install/lib gc_cpp.lo ./libgc.la )
-libtool: relink: g++-4.9  -fPIC -DPIC -shared -nostdlib /usr/lib/gcc/i586-linux-gnu/4.9/../../../i386-linux-gnu/crti.o /usr/lib/gcc/i586-linux-gnu/4.9/crtbeginS.o  .libs/gc_cpp.o   -Wl,-rpath -Wl,[...]/master.build.install/lib -L[...]/master.build.install/lib -lgc -L/usr/lib/gcc/i586-linux-gnu/4.9 -L/usr/lib/gcc/i586-linux-gnu/4.9/../../../i386-linux-gnu -L/usr/lib/gcc/i586-linux-gnu/4.9/../../../../lib -L/lib/i386-linux-gnu -L/lib/../lib -L/usr/lib/i386-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/i586-linux-gnu/4.9/../../.. -lstdc++ -lm -lc -lgcc_s /usr/lib/gcc/i586-linux-gnu/4.9/crtendS.o /usr/lib/gcc/i586-linux-gnu/4.9/../../../i386-linux-gnu/crtn.o  -g -O2   -Wl,-soname -Wl,libgccpp.so.1 -o .libs/libgccpp.so.1.0.3
+libtool: relink: g++-4.9  -fPIC -DPIC -shared -nostdlib /usr/lib/gcc/i586-gnu/4.9/../../../i386-gnu/crti.o /usr/lib/gcc/i586-gnu/4.9/crtbeginS.o  .libs/gc_cpp.o   -Wl,-rpath -Wl,[...]/master.build.install/lib -L[...]/master.build.install/lib -lgc -L/usr/lib/gcc/i586-gnu/4.9 -L/usr/lib/gcc/i586-gnu/4.9/../../../i386-gnu -L/usr/lib/gcc/i586-gnu/4.9/../../.. -L/lib/i386-gnu -L/usr/lib/i386-gnu -lstdc++ -lm -lc -lgcc_s /usr/lib/gcc/i586-gnu/4.9/crtendS.o /usr/lib/gcc/i586-gnu/4.9/../../../i386-gnu/crtn.o  -g -O2   -Wl,-soname -Wl,libgccpp.so.1 -o .libs/libgccpp.so.1.0.3
 libtool: install: /usr/bin/install -c .libs/libgccpp.so.1.0.3T [...]/master.build.install/lib/libgccpp.so.1.0.3
 libtool: install: (cd [...]/master.build.install/lib && { ln -s -f libgccpp.so.1.0.3 libgccpp.so.1 || { rm -f libgccpp.so.1 && ln -s libgccpp.so.1.0.3 libgccpp.so.1; }; })
 libtool: install: (cd [...]/master.build.install/lib && { ln -s -f libgccpp.so.1.0.3 libgccpp.so || { rm -f libgccpp.so && ln -s libgccpp.so.1.0.3 libgccpp.so; }; })
@@ -29,7 +29,7 @@ libtool: install: ranlib [...]/master.build.install/lib/libgccpp.a
 libtool: install: /usr/bin/install -c .libs/libcord.a [...]/master.build.install/lib/libcord.a
 libtool: install: chmod 644 [...]/master.build.install/lib/libcord.a
 libtool: install: ranlib [...]/master.build.install/lib/libcord.a
-libtool: finish: PATH="/home/thomas/command-i686:/home/thomas/command:/home/thomas/shared/command:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/sbin" ldconfig -n [...]/master.build.install/lib
+libtool: finish: PATH="/home/thomas/command:/home/thomas/shared/command:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/sbin" ldconfig -n [...]/master.build.install/lib
 ----------------------------------------------------------------------
 Libraries have been installed in:
    [...]/master.build.install/lib
