diff --git toolchain/logs/boehm-gc/kepler.SCHWINGE/log_test_ toolchain/logs/boehm-gc/laplace.SCHWINGE/log_test_
index c2cb48a..577e95e 100644
--- toolchain/logs/boehm-gc/kepler.SCHWINGE/log_test_
+++ toolchain/logs/boehm-gc/laplace.SCHWINGE/log_test_
@@ -125,7 +125,6 @@ PASS: disclaim_test
 PASS: disclaim_bench
 make[4]: Entering directory '[...]/master.build'
 make[5]: Entering directory '[...]/master.build'
-make[5]: Nothing to be done for 'all-am'.
 make[5]: Leaving directory '[...]/master.build'
 make[4]: Leaving directory '[...]/master.build'
 ============================================================================
