--- /dev/fd/63	2012-06-08 17:49:04.273892969 +0200
+++ /dev/fd/62	2012-06-08 17:49:04.273892969 +0200
@@ -56,652 +56,208 @@
 make  check-TESTS
 make[2]: Entering directory `[...]/master.build'
 Completed 6 tests
-Allocated 10285491 collectable objects
+Allocated 10285335 collectable objects
 Allocated 1212 uncollectable objects
-Allocated 7499999 atomic objects
-Allocated 137759 stubborn objects
-Finalized 6604/6604 objects - finalization is probably ok
-Total number of bytes allocated is 1135710310
-Final heap size is 28192768 bytes
+Allocated 7500000 atomic objects
+Allocated 137760 stubborn objects
+Finalized 6613/6613 objects - finalization is probably ok
+Total number of bytes allocated is 799534169
+Final heap size is 20905984 bytes
 Collector appears to work
-Completed 280 collections
+Completed 166 collections
 PASS: gctest
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=5, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=6, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=7, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=9, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=10, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=11, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/leak_test.c:19, sz=12, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/leak_test.c:18 [[ADDR]]
 
 PASS: leaktest
-Final heap size is 131072
+Final heap size is 1536000
 PASS: middletest
 GC_check_heap_block: found smashed heap objects:
 [ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
 GC_check_heap_block: found smashed heap objects:
 [ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
-GC_check_heap_block: found smashed heap objects:
-[ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
 [ADDR] in or near object at [ADDR] (../master/tests/smash_test.c:22, sz=40)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		main:[...]/master.build/../master/tests/smash_test.c:22 [[ADDR]]
 PASS: smashtest
 PASS: hugetest
 PASS: staticrootstest
 
 ****Chosen address [ADDR] in object
-[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
+[ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
 
 Reachable via 0 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 1 levels of pointers from offset 0 in object:
+Reachable via 1 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
-Reachable via 2 levels of pointers from offset 4 in object:
+Reachable via 2 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 3 levels of pointers from offset 4 in object:
+Reachable via 3 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 4 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 5 levels of pointers from offset 4 in object:
+Reachable via 5 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 6 levels of pointers from offset 0 in object:
+Reachable via 6 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 7 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 8 levels of pointers from offset 0 in object:
+Reachable via 8 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 9 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 10 levels of pointers from root at [ADDR]
+Reachable via 10 levels of pointers from offset 0 in object:
+[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
+
+Reachable via 11 levels of pointers from root at [ADDR]
 
 
 ****Chosen address [ADDR] in object
 [ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc_atomic+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 0 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 1 levels of pointers from offset 0 in object:
+Reachable via 1 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 2 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 3 levels of pointers from offset 4 in object:
+Reachable via 3 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
-Reachable via 4 levels of pointers from offset 4 in object:
+Reachable via 4 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 5 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 6 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 7 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 8 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 9 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 10 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 11 levels of pointers from root at [ADDR]
 
 
 ****Chosen address [ADDR] in object
-[ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc_atomic+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
+[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
 
 Reachable via 0 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
-Reachable via 1 levels of pointers from offset 4 in object:
+Reachable via 1 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
-Reachable via 2 levels of pointers from offset 0 in object:
+Reachable via 2 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 3 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 4 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 5 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 6 levels of pointers from offset 0 in object:
+Reachable via 6 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 7 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 8 levels of pointers from offset 4 in object:
+Reachable via 8 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
-
-Reachable via 9 levels of pointers from offset 0 in object:
-[ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 10 levels of pointers from offset 0 in object:
+Reachable via 9 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 11 levels of pointers from root at [ADDR]
+Reachable via 10 levels of pointers from root at [ADDR]
 
 
 ****Chosen address [ADDR] in object
 [ADDR] (../master/tests/trace_test.c:18, sz=8, PTRFREE)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc_atomic+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:18 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 0 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 1 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 2 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
 Reachable via 3 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
 
-Reachable via 4 levels of pointers from offset 4 in object:
+Reachable via 4 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
 
 Reachable via 5 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 6 levels of pointers from offset 4 in object:
+Reachable via 6 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 7 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
-Reachable via 8 levels of pointers from offset 4 in object:
+Reachable via 8 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:20 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 9 levels of pointers from offset 4 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:19 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 10 levels of pointers from offset 0 in object:
 [ADDR] (../master/tests/trace_test.c:16, sz=8, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		mktree:[...]/master.build/../master/tests/trace_test.c:17 [[ADDR]]
-		main:[...]/master.build/../master/tests/trace_test.c:29 [[ADDR]]
 
 Reachable via 11 levels of pointers from root at [ADDR]
 
 PASS: tracetest
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:12, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:12 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:12, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:12 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:12, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:12 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:12, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:12 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 Leaked composite object at [ADDR] (../master/tests/thread_leak_test.c:12, sz=4, NORMAL)
-	Call chain at allocation:
-		[...]/master.build/.libs/libgc.so.1(GC_debug_malloc+[ADDR]) [[ADDR]]
-		test:[...]/master.build/../master/tests/thread_leak_test.c:12 [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_inner_start_routine+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_call_with_stack_base+[ADDR]) [[ADDR]]
-		[...]/master.build/.libs/libgc.so.1(GC_start_routine+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libpthread.so.0(+[ADDR]) [[ADDR]]
-		/lib/i386-linux-gnu/i686/nosegneg/libc.so.6(clone+[ADDR]) [[ADDR]]
 
 PASS: threadleaktest
 PASS: initsecondarythread
