diff --git toolchain/logs/boehm-gc/kepler.SCHWINGE/log_test_ toolchain/logs/boehm-gc/laplace.SCHWINGE/log_test_
index 5b1d7ac..c3c2a46 100644
--- toolchain/logs/boehm-gc/kepler.SCHWINGE/log_test_
+++ toolchain/logs/boehm-gc/laplace.SCHWINGE/log_test_
@@ -125,7 +125,6 @@ PASS: disclaim_test
 PASS: disclaim_bench
 make[4]: Entering directory '[...]/master.build'
 make[5]: Entering directory '[...]/master.build'
-make[5]: Nothing to be done for 'all-am'.
 make[5]: Leaving directory '[...]/master.build'
 make[4]: Leaving directory '[...]/master.build'
 ============================================================================
@@ -248,7 +247,7 @@ make[3]: Leaving directory '[...]/master.build/libatomic_ops/tests'
 make  check-TESTS
 make[3]: Entering directory '[...]/master.build/libatomic_ops/tests'
 make[4]: Entering directory '[...]/master.build/libatomic_ops/tests'
-../../../master/libatomic_ops/test-driver: line 107: 21729 Aborted                 "$@" > $log_file 2>&1
+../../../master/libatomic_ops/test-driver: line 107: 27232 Aborted                 "$@" > $log_file 2>&1
 FAIL: test_atomic
 PASS: test_atomic_pthreads
 PASS: test_stack
