diff --git toolchain/logs/binutils/kepler.SCHWINGE/log_test_ toolchain/logs/binutils/coulomb.SCHWINGE/log_test_
index 5eca3b7..792149b 100644
--- toolchain/logs/binutils/kepler.SCHWINGE/log_test_
+++ toolchain/logs/binutils/coulomb.SCHWINGE/log_test_
@@ -52,7 +52,7 @@ srcdir=`cd ../../Paul_Desmond/binutils && pwd`; export srcdir; \
 	else echo "WARNING: could not find \`runtest'" 1>&2; :;\
 	fi
 WARNING: Couldn't find tool init file
-Test Run By thomas on Tue Jan  7 13:14:49 2014
+Test Run By thomas on Tue Jan  7 14:07:29 2014
 Native configuration is [ARCH]
 
 		=== binutils tests ===
@@ -131,7 +131,7 @@ rootme=`pwd`; export rootme; \
 	else echo "WARNING: could not find \`runtest'" 1>&2; :;\
 	fi
 WARNING: Couldn't find tool init file
-Test Run By thomas on Tue Jan  7 13:14:54 2014
+Test Run By thomas on Tue Jan  7 14:07:57 2014
 Native configuration is [ARCH]
 
 		=== gas tests ===
@@ -1092,8 +1092,8 @@ PASS: ver_test_4.sh
 PASS: ver_test_5.sh
 PASS: ver_test_7.sh
 PASS: ver_test_10.sh
-FAIL: relro_test.sh
-PASS: ver_matching_test.sh
+PASS: relro_test.sh
+FAIL: ver_matching_test.sh
 PASS: script_test_3.sh
 PASS: script_test_4.sh
 PASS: script_test_5.sh
@@ -1204,8 +1204,8 @@ PASS: script_test_2
 PASS: justsyms
 PASS: justsyms_exec
 PASS: binary_test
-PASS: script_test_3
-PASS: tls_phdrs_script_test
+FAIL: script_test_3
+FAIL: tls_phdrs_script_test
 PASS: tls_script_test
 PASS: script_test_11
 PASS: thin_archive_test_1
@@ -1248,7 +1248,7 @@ PASS: ifuncmain7pie
 PASS: ifuncvar
 PASS: start_lib_test
 ============================
-2 of 182 tests failed
+4 of 182 tests failed
 See testsuite/test-suite.log
 ============================
 make[6]: *** [test-suite.log] Error 1
@@ -1346,14 +1346,14 @@ srcroot=`cd ../../Paul_Desmond/ld && pwd`; export srcroot; \
 		CC="gcc-4.8" CFLAGS="-g -O2" \
 		CXX="g++-4.8" CXXFLAGS="-g -O2" \
 		CC_FOR_HOST="gcc-4.8" CFLAGS_FOR_HOST="-g -O2" \
-		OFILES="ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o plugin.o ldwrite.o ldexp.o  ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o eelf_i386.o ei386linux.o eelf32_x86_64.o " BFDLIB="../bfd/.libs/libbfd.a" \
+		OFILES="ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o plugin.o ldwrite.o ldexp.o  ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o eelf_i386.o " BFDLIB="../bfd/.libs/libbfd.a" \
 		LIBIBERTY="../libiberty/libiberty.a " LIBS="-lz -ldl " \
 		DO_COMPARE="`echo 'cmp --ignore-initial=16 $f1 $f2' | sed -e 's,\\$,,g'`" \
 		; \
 	else echo "WARNING: could not find \`runtest'" 1>&2; :;\
 	fi
 WARNING: Couldn't find tool init file
-Test Run By thomas on Tue Jan  7 13:17:03 2014
+Test Run By thomas on Tue Jan  7 14:30:52 2014
 Native configuration is [ARCH]
 
 		=== ld tests ===
@@ -1399,6 +1399,12 @@ Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-elfcomm/elfcomm.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-elfvers/vers.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-elfvsb/elfvsb.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-elfweak/elfweak.exp ...
+XPASS: ELF weak func first DSO
+XPASS: ELF weak func last DSO
+XPASS: ELF weak data first DSO
+XPASS: ELF weak data last DSO
+XPASS: ELF weak data first DSO common
+XPASS: ELF weak data last DSO common
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-fastcall/fastcall.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-frv/fdpic.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-frv/frv-elf.exp ...
@@ -1411,6 +1417,8 @@ Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ia64/ia64.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ia64/line.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ifunc/binutils.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ifunc/ifunc.exp ...
+FAIL: Common symbol override ifunc test 1a
+FAIL: Common symbol override ifunc test 1b
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-libs/libs.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-linkonce/linkonce.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-m68hc11/m68hc11.exp ...
@@ -1504,15 +1512,22 @@ Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-xtensa/xtensa.exp ...
 
 		=== ld Summary ===
 
-# of expected passes		907
-# of expected failures		8
+# of expected passes		895
+# of unexpected failures	2
+# of unexpected successes	6
+# of expected failures		12
 # of untested testcases		6
 [...]/tschwinge/Paul_Desmond.build/ld/ld-new 2.23.52.20130618
 
+make[5]: *** [check-DEJAGNU] Error 1
 make[5]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/ld'
+make[4]: *** [check-am] Error 2
 make[4]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/ld'
+make[3]: *** [check-recursive] Error 1
 make[3]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/ld'
+make[2]: *** [check] Error 2
 make[2]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/ld'
+make[1]: *** [check-ld] Error 2
 make[2]: Entering directory `[...]/tschwinge/Paul_Desmond.build/libiberty'
 make[3]: Entering directory `[...]/tschwinge/Paul_Desmond.build/libiberty/testsuite'
 gcc-4.8 -DHAVE_CONFIG_H -g -O2 -I.. -I../../../Paul_Desmond/libiberty/testsuite/../../include  -o test-demangle \
