--- toolchain/logs/binutils/kepler.SCHWINGE/log_test_	2012-12-15 23:23:52.458583339 +0100
+++ toolchain/logs/binutils/coulomb.SCHWINGE/log_test_	2012-12-15 23:23:52.458583339 +0100
@@ -52,7 +52,7 @@
 	else echo "WARNING: could not find \`runtest'" 1>&2; :;\
 	fi
 WARNING: Couldn't find tool init file
-Test Run By thomas on Sat Dec 15 15:49:15 2012
+Test Run By thomas on Sat Dec 15 19:12:37 2012
 Native configuration is [ARCH]
 
 		=== binutils tests ===
@@ -130,7 +130,7 @@ Running [...]/tschwinge/Paul_Desmond/bin
 	else echo "WARNING: could not find \`runtest'" 1>&2; :;\
 	fi
 WARNING: Couldn't find tool init file
-Test Run By thomas on Sat Dec 15 15:49:21 2012
+Test Run By thomas on Sat Dec 15 19:13:03 2012
 Native configuration is [ARCH]
 
 		=== gas tests ===
@@ -184,6 +184,8 @@ Running [...]/tschwinge/Paul_Desmond/gas
 Running [...]/tschwinge/Paul_Desmond/gas/testsuite/gas/hppa/reloc/reloc.exp ...
 Running [...]/tschwinge/Paul_Desmond/gas/testsuite/gas/hppa/unsorted/unsorted.exp ...
 Running [...]/tschwinge/Paul_Desmond/gas/testsuite/gas/i386/i386.exp ...
+WARNING: program timed out.
+FAIL: gas/i386/rept
 Running [...]/tschwinge/Paul_Desmond/gas/testsuite/gas/i386/ilp32/cfi/ilp32.exp ...
 Running [...]/tschwinge/Paul_Desmond/gas/testsuite/gas/i386/ilp32/elf/ilp32.exp ...
 Running [...]/tschwinge/Paul_Desmond/gas/testsuite/gas/i386/ilp32/ilp32.exp ...
@@ -266,12 +268,18 @@ Running [...]/tschwinge/Paul_Desmond/gas
 
 		=== gas Summary ===
 
-# of expected passes		379
+# of expected passes		378
+# of unexpected failures	1
 ../as-new 2.23.51.20121215
 
+make[4]: *** [check-DEJAGNU] Error 1
 make[4]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/gas'
+make[3]: *** [check-am] Error 2
 make[3]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/gas'
+make[2]: *** [check-recursive] Error 1
+make[2]: Target `check' not remade because of errors.
 make[2]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/gas'
+make[1]: *** [check-gas] Error 2
 make[2]: Entering directory `[...]/tschwinge/Paul_Desmond.build/gprof'
 make  check-recursive
 make[3]: Entering directory `[...]/tschwinge/Paul_Desmond.build/gprof'
@@ -308,14 +316,14 @@ Running [...]/tschwinge/Paul_Desmond/gas
 		CC="gcc-4.6" CFLAGS="-g -O2" \
 		CXX="g++-4.6" CXXFLAGS="-g -O2" \
 		CC_FOR_HOST="gcc-4.6" CFLAGS_FOR_HOST="-g -O2" \
-		OFILES="ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o plugin.o ldwrite.o ldexp.o  ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o eelf_i386.o ei386linux.o eelf32_x86_64.o " BFDLIB="../bfd/.libs/libbfd.a" \
+		OFILES="ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o plugin.o ldwrite.o ldexp.o  ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o eelf_i386.o " BFDLIB="../bfd/.libs/libbfd.a" \
 		LIBIBERTY="../libiberty/libiberty.a " LIBS="-lz -ldl " \
 		DO_COMPARE="`echo 'cmp --ignore-initial=16 $f1 $f2' | sed -e 's,\\$,,g'`" \
 		; \
 	else echo "WARNING: could not find \`runtest'" 1>&2; :;\
 	fi
 WARNING: Couldn't find tool init file
-Test Run By thomas on Sat Dec 15 15:50:12 2012
+Test Run By thomas on Sat Dec 15 19:20:36 2012
 Native configuration is [ARCH]
 
 		=== ld tests ===
@@ -372,6 +380,8 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ia64/line.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ifunc/binutils.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ifunc/ifunc.exp ...
+FAIL: Common symbol override ifunc test 1a
+FAIL: Common symbol override ifunc test 1b
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-libs/libs.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-linkonce/linkonce.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-m68hc11/m68hc11.exp ...
@@ -463,9 +473,9 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 
 		=== ld Summary ===
 
-# of expected passes		835
-# of unexpected failures	1
-# of expected failures		8
+# of expected passes		823
+# of unexpected failures	3
+# of expected failures		18
 # of untested testcases		6
 [...]/tschwinge/Paul_Desmond.build/ld/ld-new 2.23.51.20121215
 
