diff --git toolchain/logs/binutils/kepler.SCHWINGE/log_build_ toolchain/logs/binutils/coulomb.SCHWINGE/log_build_
index 6dce852..a19f3b1 100644
--- toolchain/logs/binutils/kepler.SCHWINGE/log_build_
+++ toolchain/logs/binutils/coulomb.SCHWINGE/log_build_
@@ -281,12 +281,12 @@ checking for sys/sysmp.h... no
 checking for sys/sysinfo.h... yes
 checking for machine/hal_sysinfo.h... no
 checking for sys/table.h... no
-checking for sys/sysctl.h... yes
+checking for sys/sysctl.h... no
 checking for sys/systemcfg.h... no
 checking for stdint.h... (cached) yes
 checking for stdio_ext.h... yes
 checking for process.h... no
-checking for sys/prctl.h... yes
+checking for sys/prctl.h... no
 checking for sys/wait.h that is POSIX.1 compatible... yes
 checking whether time.h and sys/time.h may both be included... yes
 checking whether errno must be declared... no
@@ -357,8 +357,8 @@ checking for vfork... yes
 checking for working fork... yes
 checking for working vfork... (cached) yes
 checking for _doprnt... no
-checking for sys_errlist... yes
-checking for sys_nerr... yes
+checking for sys_errlist... no
+checking for sys_nerr... no
 checking for sys_siglist... yes
 checking for external symbol _system_configuration... no
 checking for __fsetlocking... yes
@@ -380,7 +380,7 @@ checking for spawnvpe... no
 checking for strerror... yes
 checking for strsignal... yes
 checking for sysconf... yes
-checking for sysctl... yes
+checking for sysctl... no
 checking for sysmp... no
 checking for table... no
 checking for times... yes
@@ -465,7 +465,7 @@ checking if the linker (ld) is GNU ld... yes
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -485,7 +485,7 @@ checking if gcc-4.8 static flag -static works... yes
 checking if gcc-4.8 supports -c -o file.o... yes
 checking if gcc-4.8 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.8 linker (ld) supports shared libraries... yes
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.5 ld.so
 checking how to hardcode library paths into programs... immediate
 checking for shl_load... no
 checking for shl_load in -ldld... no
@@ -583,26 +583,26 @@ checking for zlib.h... yes
 checking sys/procfs.h usability... yes
 checking sys/procfs.h presence... yes
 checking for sys/procfs.h... yes
-checking for prstatus_t in sys/procfs.h... yes
+checking for prstatus_t in sys/procfs.h... no
 checking for prstatus32_t in sys/procfs.h... no
 checking for prstatus_t.pr_who in sys/procfs.h... no
 checking for prstatus32_t.pr_who in sys/procfs.h... no
-checking for pstatus_t in sys/procfs.h... no
+checking for pstatus_t in sys/procfs.h... yes
 checking for pxstatus_t in sys/procfs.h... no
 checking for pstatus32_t in sys/procfs.h... no
-checking for prpsinfo_t in sys/procfs.h... yes
-checking for prpsinfo_t.pr_pid in sys/procfs.h... yes
+checking for prpsinfo_t in sys/procfs.h... no
+checking for prpsinfo_t.pr_pid in sys/procfs.h... no
 checking for prpsinfo32_t in sys/procfs.h... no
 checking for prpsinfo32_t.pr_pid in sys/procfs.h... no
-checking for psinfo_t in sys/procfs.h... no
-checking for psinfo_t.pr_pid in sys/procfs.h... no
+checking for psinfo_t in sys/procfs.h... yes
+checking for psinfo_t.pr_pid in sys/procfs.h... yes
 checking for psinfo32_t in sys/procfs.h... no
 checking for psinfo32_t.pr_pid in sys/procfs.h... no
-checking for lwpstatus_t in sys/procfs.h... no
+checking for lwpstatus_t in sys/procfs.h... yes
 checking for lwpxstatus_t in sys/procfs.h... no
 checking for lwpstatus_t.pr_context in sys/procfs.h... no
-checking for lwpstatus_t.pr_reg in sys/procfs.h... no
-checking for lwpstatus_t.pr_fpreg in sys/procfs.h... no
+checking for lwpstatus_t.pr_reg in sys/procfs.h... yes
+checking for lwpstatus_t.pr_fpreg in sys/procfs.h... yes
 checking for win32_pstatus_t in sys/procfs.h... no
 checking linker --as-needed support... yes
 checking for cos in -lm... yes
@@ -1256,36 +1256,15 @@ mv -f .deps/elf-eh-frame.Tpo .deps/elf-eh-frame.Plo
 /bin/dash ./libtool --tag=CC   --mode=compile gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include  -DHAVE_bfd_elf32_i386_vec -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/tschwinge/Paul_Desmond.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT dwarf1.lo -MD -MP -MF .deps/dwarf1.Tpo -c -o dwarf1.lo ../../Paul_Desmond/bfd/dwarf1.c
 libtool: compile:  gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include -DHAVE_bfd_elf32_i386_vec -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/tschwinge/Paul_Desmond.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT dwarf1.lo -MD -MP -MF .deps/dwarf1.Tpo -c ../../Paul_Desmond/bfd/dwarf1.c -o dwarf1.o
 mv -f .deps/dwarf1.Tpo .deps/dwarf1.Plo
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd    -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/tschwinge/Paul_Desmond.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT i386linux.lo -MD -MP -MF .deps/i386linux.Tpo -c -o i386linux.lo ../../Paul_Desmond/bfd/i386linux.c
-libtool: compile:  gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd  -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/tschwinge/Paul_Desmond.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT i386linux.lo -MD -MP -MF .deps/i386linux.Tpo -c ../../Paul_Desmond/bfd/i386linux.c -o i386linux.o
-mv -f .deps/i386linux.Tpo .deps/i386linux.Plo
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd    -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/tschwinge/Paul_Desmond.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT aout32.lo -MD -MP -MF .deps/aout32.Tpo -c -o aout32.lo ../../Paul_Desmond/bfd/aout32.c
-libtool: compile:  gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd  -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/tschwinge/Paul_Desmond.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT aout32.lo -MD -MP -MF .deps/aout32.Tpo -c ../../Paul_Desmond/bfd/aout32.c -o aout32.o
-mv -f .deps/aout32.Tpo .deps/aout32.Plo
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd    -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/tschwinge/Paul_Desmond.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT pei-i386.lo -MD -MP -MF .deps/pei-i386.Tpo -c -o pei-i386.lo ../../Paul_Desmond/bfd/pei-i386.c
-libtool: compile:  gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd  -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/tschwinge/Paul_Desmond.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT pei-i386.lo -MD -MP -MF .deps/pei-i386.Tpo -c ../../Paul_Desmond/bfd/pei-i386.c -o pei-i386.o
-mv -f .deps/pei-i386.Tpo .deps/pei-i386.Plo
-rm -f peigen.c
-sed -e s/XX/pe/g < ../../Paul_Desmond/bfd/peXXigen.c > peigen.new
-mv -f peigen.new peigen.c
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd    -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/tschwinge/Paul_Desmond.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT peigen.lo -MD -MP -MF .deps/peigen.Tpo -c -o peigen.lo peigen.c
-libtool: compile:  gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd  -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/tschwinge/Paul_Desmond.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT peigen.lo -MD -MP -MF .deps/peigen.Tpo -c peigen.c -o peigen.o
-mv -f .deps/peigen.Tpo .deps/peigen.Plo
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd    -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/tschwinge/Paul_Desmond.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT cofflink.lo -MD -MP -MF .deps/cofflink.Tpo -c -o cofflink.lo ../../Paul_Desmond/bfd/cofflink.c
-libtool: compile:  gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd  -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/tschwinge/Paul_Desmond.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT cofflink.lo -MD -MP -MF .deps/cofflink.Tpo -c ../../Paul_Desmond/bfd/cofflink.c -o cofflink.o
-mv -f .deps/cofflink.Tpo .deps/cofflink.Plo
 /bin/dash ./libtool --tag=CC   --mode=compile gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include  -DHAVE_bfd_elf32_i386_vec -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/tschwinge/Paul_Desmond.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT elf32-gen.lo -MD -MP -MF .deps/elf32-gen.Tpo -c -o elf32-gen.lo ../../Paul_Desmond/bfd/elf32-gen.c
 libtool: compile:  gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include -DHAVE_bfd_elf32_i386_vec -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/tschwinge/Paul_Desmond.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT elf32-gen.lo -MD -MP -MF .deps/elf32-gen.Tpo -c ../../Paul_Desmond/bfd/elf32-gen.c -o elf32-gen.o
 mv -f .deps/elf32-gen.Tpo .deps/elf32-gen.Plo
 /bin/dash ./libtool --tag=CC   --mode=compile gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include  -DHAVE_bfd_elf32_i386_vec -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/tschwinge/Paul_Desmond.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT cpu-i386.lo -MD -MP -MF .deps/cpu-i386.Tpo -c -o cpu-i386.lo ../../Paul_Desmond/bfd/cpu-i386.c
 libtool: compile:  gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include -DHAVE_bfd_elf32_i386_vec -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/tschwinge/Paul_Desmond.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT cpu-i386.lo -MD -MP -MF .deps/cpu-i386.Tpo -c ../../Paul_Desmond/bfd/cpu-i386.c -o cpu-i386.o
 mv -f .deps/cpu-i386.Tpo .deps/cpu-i386.Plo
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd    -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/tschwinge/Paul_Desmond.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT trad-core.lo -MD -MP -MF .deps/trad-core.Tpo -c -o trad-core.lo ../../Paul_Desmond/bfd/trad-core.c
-libtool: compile:  gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/bfd  -I. -I../../Paul_Desmond/bfd -I../../Paul_Desmond/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/tschwinge/Paul_Desmond.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT trad-core.lo -MD -MP -MF .deps/trad-core.Tpo -c ../../Paul_Desmond/bfd/trad-core.c -o trad-core.o
-mv -f .deps/trad-core.Tpo .deps/trad-core.Plo
 rm -f tofiles
 f=""; \
-	for i in elf32-i386.lo elf-ifunc.lo elf-nacl.lo elf-vxworks.lo elf32.lo elf.lo elflink.lo elf-attrs.lo elf-strtab.lo elf-eh-frame.lo dwarf1.lo i386linux.lo aout32.lo pei-i386.lo peigen.lo cofflink.lo elf32-gen.lo cpu-i386.lo trad-core.lo  ; do \
+	for i in elf32-i386.lo elf-ifunc.lo elf-nacl.lo elf-vxworks.lo elf32.lo elf.lo elflink.lo elf-attrs.lo elf-strtab.lo elf-eh-frame.lo dwarf1.lo elf32-gen.lo cpu-i386.lo   ; do \
 	  case " $f " in \
 	    *" $i "*) ;; \
 	    *) f="$f $i" ;; \
@@ -1295,7 +1274,7 @@ f=""; \
 /bin/dash ../../Paul_Desmond/bfd/../move-if-change tofiles ofiles
 touch stamp-ofiles
 /bin/dash ./libtool --tag=CC   --mode=link gcc-4.8 -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2  -release `cat libtool-soversion`  -static-libstdc++ -static-libgcc  -o libbfd.la -rpath [...]/tschwinge/Paul_Desmond.build.install/lib archive.lo archures.lo bfd.lo bfdio.lo bfdwin.lo cache.lo coffgen.lo corefile.lo format.lo init.lo libbfd.lo opncls.lo reloc.lo section.lo syms.lo targets.lo hash.lo linker.lo srec.lo binary.lo tekhex.lo ihex.lo stabs.lo stab-syms.lo merge.lo dwarf2.lo simple.lo compress.lo verilog.lo `cat ofiles`   -lz 
-libtool: link: ar rc .libs/libbfd.a  archive.o archures.o bfd.o bfdio.o bfdwin.o cache.o coffgen.o corefile.o format.o init.o libbfd.o opncls.o reloc.o section.o syms.o targets.o hash.o linker.o srec.o binary.o tekhex.o ihex.o stabs.o stab-syms.o merge.o dwarf2.o simple.o compress.o verilog.o elf32-i386.o elf-ifunc.o elf-nacl.o elf-vxworks.o elf32.o elf.o elflink.o elf-attrs.o elf-strtab.o elf-eh-frame.o dwarf1.o i386linux.o aout32.o pei-i386.o peigen.o cofflink.o elf32-gen.o cpu-i386.o trad-core.o
+libtool: link: ar rc .libs/libbfd.a  archive.o archures.o bfd.o bfdio.o bfdwin.o cache.o coffgen.o corefile.o format.o init.o libbfd.o opncls.o reloc.o section.o syms.o targets.o hash.o linker.o srec.o binary.o tekhex.o ihex.o stabs.o stab-syms.o merge.o dwarf2.o simple.o compress.o verilog.o elf32-i386.o elf-ifunc.o elf-nacl.o elf-vxworks.o elf32.o elf.o elflink.o elf-attrs.o elf-strtab.o elf-eh-frame.o dwarf1.o elf32-gen.o cpu-i386.o
 libtool: link: ranlib .libs/libbfd.a
 libtool: link: ( cd ".libs" && rm -f "libbfd.la" && ln -s "../libbfd.la" "libbfd.la" )
 libtooldir=`/bin/dash ./libtool --config | sed -n -e 's/^objdir=//p'`; \
@@ -1362,7 +1341,7 @@ checking if the linker (ld) is GNU ld... yes
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -1382,7 +1361,7 @@ checking if gcc-4.8 static flag -static works... yes
 checking if gcc-4.8 supports -c -o file.o... yes
 checking if gcc-4.8 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.8 linker (ld) supports shared libraries... yes
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.5 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether stripping libraries is possible... yes
 checking if libtool supports shared libraries... yes
@@ -1543,7 +1522,7 @@ checking if the linker (ld) is GNU ld... yes
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -1564,7 +1543,7 @@ checking if gcc-4.8 supports -c -o file.o... yes
 checking if gcc-4.8 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.8 linker (ld) supports shared libraries... yes
 checking whether -lc should be explicitly linked in... no
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.5 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether stripping libraries is possible... yes
 checking if libtool supports shared libraries... yes
@@ -1713,7 +1692,7 @@ checking if the linker (ld) is GNU ld... yes
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -1734,7 +1713,7 @@ checking if gcc-4.8 supports -c -o file.o... yes
 checking if gcc-4.8 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.8 linker (ld) supports shared libraries... yes
 checking whether -lc should be explicitly linked in... no
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.5 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether stripping libraries is possible... yes
 checking if libtool supports shared libraries... yes
@@ -2315,7 +2294,7 @@ checking for mmap... yes
 checking for pread... yes
 checking for ftruncate... yes
 checking for ffsll... yes
-checking mremap with MREMAP_MAYMOVE... yes
+checking mremap with MREMAP_MAYMOVE... no
 checking for library containing zlibVersion... -lz
 checking zlib.h usability... yes
 checking zlib.h presence... yes
@@ -2351,7 +2330,7 @@ checking dlfcn.h presence... yes
 checking for dlfcn.h... yes
 checking for mallinfo... yes
 checking for posix_fallocate... yes
-checking for fallocate... yes
+checking for fallocate... no
 checking for readv... yes
 checking for sysconf... yes
 checking for times... yes
@@ -2513,8 +2492,10 @@ g++-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/gold  -I../../Paul_Desmond/gold
 mv -f .deps/workqueue-threads.Tpo .deps/workqueue-threads.Po
 gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/gold  -I../../Paul_Desmond/gold -I../../Paul_Desmond/gold/../include -I../../Paul_Desmond/gold/../elfcpp -DLOCALEDIR="\"[...]/tschwinge/Paul_Desmond.build.install/share/locale\"" -DBINDIR="\"[...]/tschwinge/Paul_Desmond.build.install/bin\"" -DTOOLBINDIR="\"[...]/tschwinge/Paul_Desmond.build.install/[ARCH]/bin\"" -DTOOLLIBDIR="\"[...]/tschwinge/Paul_Desmond.build.install/[ARCH]/lib\""   -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -frandom-seed=yyscript.o -g -O2 -MT yyscript.o -MD -MP -MF .deps/yyscript.Tpo -c -o yyscript.o yyscript.c
 mv -f .deps/yyscript.Tpo .deps/yyscript.Po
+gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/gold  -I../../Paul_Desmond/gold -I../../Paul_Desmond/gold/../include -I../../Paul_Desmond/gold/../elfcpp -DLOCALEDIR="\"[...]/tschwinge/Paul_Desmond.build.install/share/locale\"" -DBINDIR="\"[...]/tschwinge/Paul_Desmond.build.install/bin\"" -DTOOLBINDIR="\"[...]/tschwinge/Paul_Desmond.build.install/[ARCH]/bin\"" -DTOOLLIBDIR="\"[...]/tschwinge/Paul_Desmond.build.install/[ARCH]/lib\""   -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -frandom-seed=mremap.o -g -O2 -MT mremap.o -MD -MP -MF .deps/mremap.Tpo -c -o mremap.o ../../Paul_Desmond/gold/mremap.c
+mv -f .deps/mremap.Tpo .deps/mremap.Po
 rm -f libgold.a
-ar cru libgold.a archive.o attributes.o binary.o common.o compressed_output.o copy-relocs.o cref.o defstd.o descriptors.o dirsearch.o dynobj.o dwarf_reader.o ehframe.o errors.o expression.o fileread.o gc.o gdb-index.o gold.o gold-threads.o icf.o incremental.o int_encoding.o layout.o mapfile.o merge.o nacl.o object.o options.o output.o parameters.o plugin.o readsyms.o reduced_debug_output.o reloc.o resolve.o script-sections.o script.o stringpool.o symtab.o target.o target-select.o timer.o version.o workqueue.o workqueue-threads.o  yyscript.o  
+ar cru libgold.a archive.o attributes.o binary.o common.o compressed_output.o copy-relocs.o cref.o defstd.o descriptors.o dirsearch.o dynobj.o dwarf_reader.o ehframe.o errors.o expression.o fileread.o gc.o gdb-index.o gold.o gold-threads.o icf.o incremental.o int_encoding.o layout.o mapfile.o merge.o nacl.o object.o options.o output.o parameters.o plugin.o readsyms.o reduced_debug_output.o reloc.o resolve.o script-sections.o script.o stringpool.o symtab.o target.o target-select.o timer.o version.o workqueue.o workqueue-threads.o  yyscript.o  mremap.o
 ranlib libgold.a
 g++-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/gold  -I../../Paul_Desmond/gold -I../../Paul_Desmond/gold/../include -I../../Paul_Desmond/gold/../elfcpp -DLOCALEDIR="\"[...]/tschwinge/Paul_Desmond.build.install/share/locale\"" -DBINDIR="\"[...]/tschwinge/Paul_Desmond.build.install/bin\"" -DTOOLBINDIR="\"[...]/tschwinge/Paul_Desmond.build.install/[ARCH]/bin\"" -DTOOLLIBDIR="\"[...]/tschwinge/Paul_Desmond.build.install/[ARCH]/lib\""   -W -Wall    -Werror -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -frandom-seed=dwp.o -g -O2 -MT dwp.o -MD -MP -MF .deps/dwp.Tpo -c -o dwp.o ../../Paul_Desmond/gold/dwp.cc
 mv -f .deps/dwp.Tpo .deps/dwp.Po
@@ -2596,7 +2577,7 @@ checking if the linker (ld) is GNU ld... yes
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -2617,7 +2598,7 @@ checking if gcc-4.8 supports -c -o file.o... yes
 checking if gcc-4.8 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.8 linker (ld) supports shared libraries... yes
 checking whether -lc should be explicitly linked in... no
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.5 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether stripping libraries is possible... yes
 checking if libtool supports shared libraries... yes
@@ -2847,7 +2828,7 @@ checking if the linker (ld) is GNU ld... yes
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -2868,7 +2849,7 @@ checking if gcc-4.8 supports -c -o file.o... yes
 checking if gcc-4.8 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.8 linker (ld) supports shared libraries... yes
 checking whether -lc should be explicitly linked in... no
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.5 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether stripping libraries is possible... yes
 checking if libtool supports shared libraries... yes
@@ -2884,7 +2865,7 @@ checking if g++-4.8 static flag -static works... yes
 checking if g++-4.8 supports -c -o file.o... yes
 checking if g++-4.8 supports -c -o file.o... (cached) yes
 checking whether the g++-4.8 linker (ld) supports shared libraries... yes
-checking dynamic linker characteristics... (cached) GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.5 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether NLS is requested... yes
 checking for catalogs to be installed...  bg da es fi fr ga id it ja sv tr uk vi zh_CN zh_TW
@@ -2970,13 +2951,13 @@ make[2]: Entering directory `[...]/tschwinge/Paul_Desmond.build/ld'
 /bin/dash ../../Paul_Desmond/ld/../ylwrap ../../Paul_Desmond/ld/ldgram.y y.tab.c ldgram.c y.tab.h ldgram.h y.output ldgram.output -- bison -y -d 
 updating ldgram.h
 (echo "/* This file is automatically generated.  DO NOT EDIT! */";\
-	for f in `echo " " eelf_i386.o ei386linux.o eelf32_x86_64.o "" \
+	for f in `echo " " eelf_i386.o "" \
 	 | sed -e 's/ e/ ld/g' -e 's/ ld/ /g' -e 's/[.]o//g'`; do \
 	    echo "extern ld_emulation_xfer_type ld_${f}_emulation;"; \
 	done;\
 	echo "";\
 	echo "#define EMULATION_LIST \\";\
-	for f in `echo " " eelf_i386.o ei386linux.o eelf32_x86_64.o "" \
+	for f in `echo " " eelf_i386.o "" \
 	 | sed -e 's/ e/ ld/g' -e 's/ ld/ /g' -e 's/[.]o//g'`; do \
 	    echo "  &ld_${f}_emulation, \\"; \
 	done;\
@@ -3099,17 +3080,11 @@ mv -f .deps/ldcref.Tpo .deps/ldcref.Po
 gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/ld  -I. -I../../Paul_Desmond/ld -I../bfd -I../../Paul_Desmond/ld/../bfd -I../../Paul_Desmond/ld/../include  -g -O2 -DENABLE_PLUGINS -DLOCALEDIR="\"[...]/tschwinge/Paul_Desmond.build.install/share/locale\""  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT plugin.o -MD -MP -MF .deps/plugin.Tpo -c -o plugin.o ../../Paul_Desmond/ld/plugin.c
 mv -f .deps/plugin.Tpo .deps/plugin.Po
 cp ../../Paul_Desmond/ld/emultempl/astring.sed stringify.sed
-LIB_PATH='' /bin/dash ../../Paul_Desmond/ld/genscripts.sh "../../Paul_Desmond/ld" "[...]/tschwinge/Paul_Desmond.build.install/lib" "[...]/tschwinge/Paul_Desmond.build.install" "[...]/tschwinge/Paul_Desmond.build.install" [ARCH] [ARCH] [ARCH] "elf_i386 elf32_x86_64" "/usr/local/lib /lib /usr/lib" yes yes elf_i386 "[ARCH]"
+LIB_PATH='' /bin/dash ../../Paul_Desmond/ld/genscripts.sh "../../Paul_Desmond/ld" "[...]/tschwinge/Paul_Desmond.build.install/lib" "[...]/tschwinge/Paul_Desmond.build.install" "[...]/tschwinge/Paul_Desmond.build.install" [ARCH] [ARCH] [ARCH] "elf_i386" "/usr/local/lib /lib /usr/lib" yes yes elf_i386 "[ARCH]"
 gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/ld  -I. -I../../Paul_Desmond/ld -I../bfd -I../../Paul_Desmond/ld/../bfd -I../../Paul_Desmond/ld/../include  -g -O2 -DENABLE_PLUGINS -DLOCALEDIR="\"[...]/tschwinge/Paul_Desmond.build.install/share/locale\""  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT eelf_i386.o -MD -MP -MF .deps/eelf_i386.Tpo -c -o eelf_i386.o eelf_i386.c
 mv -f .deps/eelf_i386.Tpo .deps/eelf_i386.Po
-LIB_PATH='' /bin/dash ../../Paul_Desmond/ld/genscripts.sh "../../Paul_Desmond/ld" "[...]/tschwinge/Paul_Desmond.build.install/lib" "[...]/tschwinge/Paul_Desmond.build.install" "[...]/tschwinge/Paul_Desmond.build.install" [ARCH] [ARCH] [ARCH] "elf_i386 elf32_x86_64" "/usr/local/lib /lib /usr/lib" yes yes i386linux "[ARCH]aout"
-gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/ld  -I. -I../../Paul_Desmond/ld -I../bfd -I../../Paul_Desmond/ld/../bfd -I../../Paul_Desmond/ld/../include  -g -O2 -DENABLE_PLUGINS -DLOCALEDIR="\"[...]/tschwinge/Paul_Desmond.build.install/share/locale\""  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT ei386linux.o -MD -MP -MF .deps/ei386linux.Tpo -c -o ei386linux.o ei386linux.c
-mv -f .deps/ei386linux.Tpo .deps/ei386linux.Po
-LIB_PATH='' /bin/dash ../../Paul_Desmond/ld/genscripts.sh "../../Paul_Desmond/ld" "[...]/tschwinge/Paul_Desmond.build.install/lib" "[...]/tschwinge/Paul_Desmond.build.install" "[...]/tschwinge/Paul_Desmond.build.install" [ARCH] [ARCH] [ARCH] "elf_i386 elf32_x86_64" "/usr/local/lib /lib /usr/lib" yes yes elf32_x86_64 "[ARCH]"
-gcc-4.8 -DHAVE_CONFIG_H -I. -I../../Paul_Desmond/ld  -I. -I../../Paul_Desmond/ld -I../bfd -I../../Paul_Desmond/ld/../bfd -I../../Paul_Desmond/ld/../include  -g -O2 -DENABLE_PLUGINS -DLOCALEDIR="\"[...]/tschwinge/Paul_Desmond.build.install/share/locale\""  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT eelf32_x86_64.o -MD -MP -MF .deps/eelf32_x86_64.Tpo -c -o eelf32_x86_64.o eelf32_x86_64.c
-mv -f .deps/eelf32_x86_64.Tpo .deps/eelf32_x86_64.Po
-/bin/dash ./libtool --tag=CC   --mode=link gcc-4.8 -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2  -static-libstdc++ -static-libgcc  -o ld-new ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o ldwrite.o ldexp.o ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o plugin.o eelf_i386.o ei386linux.o eelf32_x86_64.o  ../bfd/libbfd.la ../libiberty/libiberty.a  -lz -ldl 
-libtool: link: gcc-4.8 -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -static-libstdc++ -static-libgcc -o ld-new ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o ldwrite.o ldexp.o ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o plugin.o eelf_i386.o ei386linux.o eelf32_x86_64.o  ../bfd/.libs/libbfd.a ../libiberty/libiberty.a -lz -ldl
+/bin/dash ./libtool --tag=CC   --mode=link gcc-4.8 -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2  -static-libstdc++ -static-libgcc  -o ld-new ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o ldwrite.o ldexp.o ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o plugin.o eelf_i386.o  ../bfd/libbfd.la ../libiberty/libiberty.a  -lz -ldl 
+libtool: link: gcc-4.8 -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -static-libstdc++ -static-libgcc -o ld-new ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o ldwrite.o ldexp.o ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o plugin.o eelf_i386.o  ../bfd/.libs/libbfd.a ../libiberty/libiberty.a -lz -ldl
 touch ld.1
 perl ../../Paul_Desmond/ld/../etc/texi2pod.pl -I ../../Paul_Desmond/ld -I ../../Paul_Desmond/ld/../bfd/doc -I ../bfd/doc -I ../../Paul_Desmond/ld/../libiberty -Dman < ../../Paul_Desmond/ld/ld.texinfo > ld.pod
 (pod2man --center="GNU Development Tools" --release="binutils-2.24.51" --section=1 ld.pod | \
