--- toolchain/logs/binutils/kepler.SCHWINGE/log_test_	2013-06-18 13:47:49.942880802 +0200
+++ toolchain/logs/binutils/coulomb.SCHWINGE/log_test_	2013-06-18 13:47:49.946880802 +0200
@@ -11,7 +11,6 @@
 make[4]: Nothing to be done for `check'.
 make[4]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/bfd/po'
 make[4]: Entering directory `[...]/tschwinge/Paul_Desmond.build/bfd'
-make[4]: Nothing to be done for `check-am'.
 make[4]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/bfd'
 make[3]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/bfd'
 make[2]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/bfd'
@@ -52,7 +51,7 @@
 	else echo "WARNING: could not find \`runtest'" 1>&2; :;\
 	fi
 WARNING: Couldn't find tool init file
-Test Run By thomas on Tue Jun 18 13:30:53 2013
+Test Run By thomas on Tue Jun 18 11:56:26 2013
 Native configuration is [ARCH]
 
 		=== binutils tests ===
@@ -131,7 +130,7 @@ Running [...]/tschwinge/Paul_Desmond/bin
 	else echo "WARNING: could not find \`runtest'" 1>&2; :;\
 	fi
 WARNING: Couldn't find tool init file
-Test Run By thomas on Tue Jun 18 13:31:00 2013
+Test Run By thomas on Tue Jun 18 11:56:54 2013
 Native configuration is [ARCH]
 
 		=== gas tests ===
@@ -307,7 +306,6 @@ Running [...]/tschwinge/Paul_Desmond/gas
 make[4]: Nothing to be done for `all'.
 make[4]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/bfd/po'
 make[4]: Entering directory `[...]/tschwinge/Paul_Desmond.build/bfd'
-make[4]: Nothing to be done for `all-am'.
 make[4]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/bfd'
 make[3]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/bfd'
 make[2]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/bfd'
@@ -336,7 +334,6 @@ Running [...]/tschwinge/Paul_Desmond/gas
 make[4]: Nothing to be done for `all'.
 make[4]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/gas/po'
 make[4]: Entering directory `[...]/tschwinge/Paul_Desmond.build/gas'
-make[4]: Nothing to be done for `all-am'.
 make[4]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/gas'
 make[3]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/gas'
 make[2]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/gas'
@@ -1084,8 +1081,8 @@ Running [...]/tschwinge/Paul_Desmond/gas
 PASS: ver_test_5.sh
 PASS: ver_test_7.sh
 PASS: ver_test_10.sh
-FAIL: relro_test.sh
-PASS: ver_matching_test.sh
+PASS: relro_test.sh
+FAIL: ver_matching_test.sh
 PASS: script_test_3.sh
 PASS: script_test_4.sh
 PASS: script_test_5.sh
@@ -1195,8 +1192,8 @@ Running [...]/tschwinge/Paul_Desmond/gas
 PASS: justsyms
 PASS: justsyms_exec
 PASS: binary_test
-PASS: script_test_3
-PASS: tls_phdrs_script_test
+FAIL: script_test_3
+FAIL: tls_phdrs_script_test
 PASS: tls_script_test
 PASS: script_test_11
 PASS: thin_archive_test_1
@@ -1239,7 +1236,7 @@ Running [...]/tschwinge/Paul_Desmond/gas
 PASS: ifuncvar
 PASS: start_lib_test
 ============================
-2 of 180 tests failed
+4 of 180 tests failed
 See testsuite/test-suite.log
 ============================
 make[6]: *** [test-suite.log] Error 1
@@ -1337,14 +1334,14 @@ Running [...]/tschwinge/Paul_Desmond/gas
 		CC="gcc-4.8" CFLAGS="-g -O2" \
 		CXX="g++-4.8" CXXFLAGS="-g -O2" \
 		CC_FOR_HOST="gcc-4.8" CFLAGS_FOR_HOST="-g -O2" \
-		OFILES="ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o plugin.o ldwrite.o ldexp.o  ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o eelf_i386.o ei386linux.o eelf32_x86_64.o " BFDLIB="../bfd/.libs/libbfd.a" \
+		OFILES="ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o plugin.o ldwrite.o ldexp.o  ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o eelf_i386.o " BFDLIB="../bfd/.libs/libbfd.a" \
 		LIBIBERTY="../libiberty/libiberty.a " LIBS="-lz -ldl " \
 		DO_COMPARE="`echo 'cmp --ignore-initial=16 $f1 $f2' | sed -e 's,\\$,,g'`" \
 		; \
 	else echo "WARNING: could not find \`runtest'" 1>&2; :;\
 	fi
 WARNING: Couldn't find tool init file
-Test Run By thomas on Tue Jun 18 13:33:22 2013
+Test Run By thomas on Tue Jun 18 12:20:39 2013
 Native configuration is [ARCH]
 
 		=== ld tests ===
@@ -1402,6 +1399,8 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ia64/line.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ifunc/binutils.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ifunc/ifunc.exp ...
+FAIL: Common symbol override ifunc test 1a
+FAIL: Common symbol override ifunc test 1b
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-libs/libs.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-linkonce/linkonce.exp ...
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-m68hc11/m68hc11.exp ...
@@ -1495,15 +1494,21 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 
 		=== ld Summary ===
 
-# of expected passes		907
-# of expected failures		8
+# of expected passes		895
+# of unexpected failures	2
+# of expected failures		18
 # of untested testcases		6
 [...]/tschwinge/Paul_Desmond.build/ld/ld-new 2.23.52.20130527
 
+make[5]: *** [check-DEJAGNU] Error 1
 make[5]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/ld'
+make[4]: *** [check-am] Error 2
 make[4]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/ld'
+make[3]: *** [check-recursive] Error 1
 make[3]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/ld'
+make[2]: *** [check] Error 2
 make[2]: Leaving directory `[...]/tschwinge/Paul_Desmond.build/ld'
+make[1]: *** [check-ld] Error 2
 make[2]: Entering directory `[...]/tschwinge/Paul_Desmond.build/libiberty'
 make[3]: Entering directory `[...]/tschwinge/Paul_Desmond.build/libiberty/testsuite'
 gcc-4.8 -DHAVE_CONFIG_H -g -O2 -I.. -I../../../Paul_Desmond/libiberty/testsuite/../../include  -o test-demangle \
