diff --git toolchain/logs/binutils/kepler.SCHWINGE/log_install_ toolchain/logs/binutils/coulomb.SCHWINGE/log_install_
index 80c57b8..a0394d1 100644
--- toolchain/logs/binutils/kepler.SCHWINGE/log_install_
+++ toolchain/logs/binutils/coulomb.SCHWINGE/log_install_
@@ -70,7 +70,6 @@ libtool: install: /usr/bin/install -c .libs/libbfd.lai [...]/tschwinge/Paul_Desm
 libtool: install: /usr/bin/install -c .libs/libbfd.a [...]/tschwinge/Paul_Desmond.build.install/lib/libbfd.a
 libtool: install: chmod 644 [...]/tschwinge/Paul_Desmond.build.install/lib/libbfd.a
 libtool: install: ranlib [...]/tschwinge/Paul_Desmond.build.install/lib/libbfd.a
-libtool: finish: PATH="/home/thomas/command-i686:/home/thomas/command:/home/thomas/shared/command:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/sbin" ldconfig -n [...]/tschwinge/Paul_Desmond.build.install/lib
 ----------------------------------------------------------------------
 Libraries have been installed in:
    [...]/tschwinge/Paul_Desmond.build.install/lib
@@ -106,7 +105,6 @@ libtool: install: /usr/bin/install -c .libs/libopcodes.lai [...]/tschwinge/Paul_
 libtool: install: /usr/bin/install -c .libs/libopcodes.a [...]/tschwinge/Paul_Desmond.build.install/lib/libopcodes.a
 libtool: install: chmod 644 [...]/tschwinge/Paul_Desmond.build.install/lib/libopcodes.a
 libtool: install: ranlib [...]/tschwinge/Paul_Desmond.build.install/lib/libopcodes.a
-libtool: finish: PATH="/home/thomas/command-i686:/home/thomas/command:/home/thomas/shared/command:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/sbin" ldconfig -n [...]/tschwinge/Paul_Desmond.build.install/lib
 ----------------------------------------------------------------------
 Libraries have been installed in:
    [...]/tschwinge/Paul_Desmond.build.install/lib
