--- open_issues/binutils/sum_linux	2012-02-15 23:36:39.000000000 +0100
+++ open_issues/binutils/sum_hurd	2012-02-15 23:36:51.000000000 +0100
@@ -1,5 +1,5 @@
-Test Run By thomas on Wed Feb 15 23:08:06 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Wed Feb 15 23:25:26 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== binutils tests ===
 
@@ -123,8 +123,8 @@ Running [...]/hurd/master/binutils/tests
 
 # of expected passes		91
 # of unsupported tests		3
-Test Run By thomas on Wed Feb 15 23:08:31 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Wed Feb 15 23:27:26 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== ld tests ===
 
@@ -336,10 +336,10 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: PIE init array
 PASS: PIE fini array
 PASS: PIE init array mixed
-PASS: static preinit array
-PASS: static init array
-PASS: static fini array
-PASS: static init array mixed
+XFAIL: static preinit array
+XFAIL: static init array
+XFAIL: static fini array
+XFAIL: static init array mixed
 Running [...]/hurd/master/ld/testsuite/ld-elf/exclude.exp ...
 PASS: ld link shared library
 PASS: ld export symbols from archive
@@ -597,8 +597,8 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: ELF DSO weak func last DSO
 PASS: ELF weak func first
 PASS: ELF weak func last
-PASS: ELF weak func first DSO
-PASS: ELF weak func last DSO
+XFAIL: ELF weak func first DSO
+XFAIL: ELF weak func last DSO
 PASS: ELF DSO weak data first
 PASS: ELF DSO weak data last
 PASS: ELF DSO weak data first DSO
@@ -609,10 +609,10 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: ELF weak data last
 PASS: ELF weak data first common
 PASS: ELF weak data last common
-PASS: ELF weak data first DSO
-PASS: ELF weak data last DSO
-PASS: ELF weak data first DSO common
-PASS: ELF weak data last DSO common
+XFAIL: ELF weak data first DSO
+XFAIL: ELF weak data last DSO
+XFAIL: ELF weak data first DSO common
+XFAIL: ELF weak data last DSO common
 PASS: ELF DSO small bar (size)
 PASS: ELF DSO foo with small bar (size)
 PASS: ELF DSO big bar (size)
@@ -948,13 +948,13 @@ Running [...]/hurd/master/ld/testsuite/l
 
 		=== ld Summary ===
 
-# of expected passes		677
-# of expected failures		8
+# of expected passes		667
+# of expected failures		18
 # of untested testcases		6
 [...]/hurd/master.build/ld/ld-new 2.22.52.20120215
 
-Test Run By thomas on Wed Feb 15 23:08:12 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Wed Feb 15 23:25:53 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== gas tests ===
 
