--- toolchain/logs/binutils/linux/sum	2012-06-07 12:17:05.321788410 +0200
+++ toolchain/logs/binutils/hurd/sum	2012-06-07 12:16:56.709788667 +0200
@@ -1,5 +1,5 @@
-Test Run By thomas on Thu Jun  7 12:14:25 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Thu Jun  7 02:07:39 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== binutils tests ===
 
@@ -123,8 +123,8 @@ Running [...]/hurd/master/binutils/tests
 
 # of expected passes		91
 # of unsupported tests		3
-Test Run By thomas on Thu Jun  7 12:14:51 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Thu Jun  7 02:10:36 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== ld tests ===
 
@@ -337,10 +337,10 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: PIE init array
 PASS: PIE fini array
 PASS: PIE init array mixed
-PASS: static preinit array
-PASS: static init array
-PASS: static fini array
-PASS: static init array mixed
+XFAIL: static preinit array
+XFAIL: static init array
+XFAIL: static fini array
+XFAIL: static init array mixed
 Running [...]/hurd/master/ld/testsuite/ld-elf/exclude.exp ...
 PASS: ld link shared library
 PASS: ld export symbols from archive
@@ -598,8 +598,8 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: ELF DSO weak func last DSO
 PASS: ELF weak func first
 PASS: ELF weak func last
-PASS: ELF weak func first DSO
-PASS: ELF weak func last DSO
+XFAIL: ELF weak func first DSO
+XFAIL: ELF weak func last DSO
 PASS: ELF DSO weak data first
 PASS: ELF DSO weak data last
 PASS: ELF DSO weak data first DSO
@@ -610,10 +610,10 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: ELF weak data last
 PASS: ELF weak data first common
 PASS: ELF weak data last common
-PASS: ELF weak data first DSO
-PASS: ELF weak data last DSO
-PASS: ELF weak data first DSO common
-PASS: ELF weak data last DSO common
+XFAIL: ELF weak data first DSO
+XFAIL: ELF weak data last DSO
+XFAIL: ELF weak data first DSO common
+XFAIL: ELF weak data last DSO common
 PASS: ELF DSO small bar (size)
 PASS: ELF DSO foo with small bar (size)
 PASS: ELF DSO big bar (size)
@@ -955,13 +955,13 @@ Running [...]/hurd/master/ld/testsuite/l
 
 		=== ld Summary ===
 
-# of expected passes		684
-# of expected failures		8
+# of expected passes		674
+# of expected failures		18
 # of untested testcases		6
 [...]/hurd/master.build/ld/ld-new 2.22.52.20120504
 
-Test Run By thomas on Thu Jun  7 12:14:31 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Thu Jun  7 02:08:28 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== gas tests ===
 
