diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/binutils/binutils.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/binutils/binutils.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/binutils/binutils.sum	2013-06-18 13:46:42.878876647 +0200
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/binutils/binutils.sum	2013-06-18 13:16:55.086765968 +0200
@@ -1,5 +1,5 @@
-Test Run By thomas on Tue Jun 18 13:30:53 2013
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Tue Jun 18 11:56:26 2013
+Native configuration is i686-unknown-gnu0.3
 
 		=== binutils tests ===
 
diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/gas/testsuite/gas.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/gas/testsuite/gas.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/gas/testsuite/gas.sum	2013-06-18 13:46:42.882876646 +0200
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/gas/testsuite/gas.sum	2013-06-18 13:16:55.102765971 +0200
@@ -1,5 +1,5 @@
-Test Run By thomas on Tue Jun 18 13:31:00 2013
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Tue Jun 18 11:56:54 2013
+Native configuration is i686-unknown-gnu0.3
 
 		=== gas tests ===
 
diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/ld/ld.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/ld/ld.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/ld/ld.sum	2013-06-18 13:46:42.882876646 +0200
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/ld/ld.sum	2013-06-18 13:16:55.134765972 +0200
@@ -1,5 +1,5 @@
-Test Run By thomas on Tue Jun 18 13:33:22 2013
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Tue Jun 18 12:20:39 2013
+Native configuration is i686-unknown-gnu0.3
 
 		=== ld tests ===
 
@@ -242,10 +242,10 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 PASS: PIE fini array
 PASS: PIE init array mixed
 PASS: PIE PR ld/14525
-PASS: static preinit array
-PASS: static init array
-PASS: static fini array
-PASS: static init array mixed
+XFAIL: static preinit array
+XFAIL: static init array
+XFAIL: static fini array
+XFAIL: static init array mixed
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-elf/exclude.exp ...
 PASS: ld link shared library
 PASS: ld export symbols from archive
@@ -561,8 +561,8 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 PASS: ELF DSO weak func last DSO
 PASS: ELF weak func first
 PASS: ELF weak func last
-PASS: ELF weak func first DSO
-PASS: ELF weak func last DSO
+XFAIL: ELF weak func first DSO
+XFAIL: ELF weak func last DSO
 PASS: ELF DSO weak data first
 PASS: ELF DSO weak data last
 PASS: ELF DSO weak data first DSO
@@ -573,10 +573,10 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 PASS: ELF weak data last
 PASS: ELF weak data first common
 PASS: ELF weak data last common
-PASS: ELF weak data first DSO
-PASS: ELF weak data last DSO
-PASS: ELF weak data first DSO common
-PASS: ELF weak data last DSO common
+XFAIL: ELF weak data first DSO
+XFAIL: ELF weak data last DSO
+XFAIL: ELF weak data first DSO common
+XFAIL: ELF weak data last DSO common
 PASS: ELF DSO small bar (size)
 PASS: ELF DSO foo with small bar (size)
 PASS: ELF DSO big bar (size)
@@ -672,8 +672,8 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ifunc/ifunc.exp ...
 PASS: Building ifunc binaries
 PASS: Checking ifunc binaries
-PASS: Common symbol override ifunc test 1a
-PASS: Common symbol override ifunc test 1b
+FAIL: Common symbol override ifunc test 1a
+FAIL: Common symbol override ifunc test 1b
 PASS: ld-ifunc/ifunc-1-local-x86
 PASS: ld-ifunc/ifunc-1-x86
 PASS: ld-ifunc/ifunc-10-i386
@@ -1067,8 +1067,9 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 
 		=== ld Summary ===
 
-# of expected passes		907
-# of expected failures		8
+# of expected passes		895
+# of unexpected failures	2
+# of expected failures		18
 # of untested testcases		6
 [...]/tschwinge/Paul_Desmond.build/ld/ld-new 2.23.52.20130527
 
