diff --git toolchain/logs/binutils/kepler.SCHWINGE/test/binutils/binutils.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/binutils/binutils.sum
index 6b73857..4ead36f 100644
--- toolchain/logs/binutils/kepler.SCHWINGE/test/binutils/binutils.sum
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/binutils/binutils.sum
@@ -1,5 +1,5 @@
-Test Run By thomas on Tue Jan  7 13:14:49 2014
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Tue Jan  7 14:07:29 2014
+Native configuration is i686-unknown-gnu0.5
 
 		=== binutils tests ===
 
diff --git toolchain/logs/binutils/kepler.SCHWINGE/test/gas/testsuite/gas.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/gas/testsuite/gas.sum
index b89fa4c..0d7bdda 100644
--- toolchain/logs/binutils/kepler.SCHWINGE/test/gas/testsuite/gas.sum
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/gas/testsuite/gas.sum
@@ -1,5 +1,5 @@
-Test Run By thomas on Tue Jan  7 13:14:54 2014
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Tue Jan  7 14:07:57 2014
+Native configuration is i686-unknown-gnu0.5
 
 		=== gas tests ===
 
diff --git toolchain/logs/binutils/kepler.SCHWINGE/test/ld/ld.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/ld/ld.sum
index eb1d5f9..1866d22 100644
--- toolchain/logs/binutils/kepler.SCHWINGE/test/ld/ld.sum
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/ld/ld.sum
@@ -1,5 +1,5 @@
-Test Run By thomas on Tue Jan  7 13:17:03 2014
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Tue Jan  7 14:30:52 2014
+Native configuration is i686-unknown-gnu0.5
 
 		=== ld tests ===
 
@@ -242,10 +242,10 @@ PASS: PIE init array
 PASS: PIE fini array
 PASS: PIE init array mixed
 PASS: PIE PR ld/14525
-PASS: static preinit array
-PASS: static init array
-PASS: static fini array
-PASS: static init array mixed
+XFAIL: static preinit array
+XFAIL: static init array
+XFAIL: static fini array
+XFAIL: static init array mixed
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-elf/exclude.exp ...
 PASS: ld link shared library
 PASS: ld export symbols from archive
@@ -561,8 +561,8 @@ PASS: ELF DSO weak func first DSO
 PASS: ELF DSO weak func last DSO
 PASS: ELF weak func first
 PASS: ELF weak func last
-PASS: ELF weak func first DSO
-PASS: ELF weak func last DSO
+XPASS: ELF weak func first DSO
+XPASS: ELF weak func last DSO
 PASS: ELF DSO weak data first
 PASS: ELF DSO weak data last
 PASS: ELF DSO weak data first DSO
@@ -573,10 +573,10 @@ PASS: ELF weak data first
 PASS: ELF weak data last
 PASS: ELF weak data first common
 PASS: ELF weak data last common
-PASS: ELF weak data first DSO
-PASS: ELF weak data last DSO
-PASS: ELF weak data first DSO common
-PASS: ELF weak data last DSO common
+XPASS: ELF weak data first DSO
+XPASS: ELF weak data last DSO
+XPASS: ELF weak data first DSO common
+XPASS: ELF weak data last DSO common
 PASS: ELF DSO small bar (size)
 PASS: ELF DSO foo with small bar (size)
 PASS: ELF DSO big bar (size)
@@ -672,8 +672,8 @@ PASS: objcopy  (ifunc-4-local-x86)
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ifunc/ifunc.exp ...
 PASS: Building ifunc binaries
 PASS: Checking ifunc binaries
-PASS: Common symbol override ifunc test 1a
-PASS: Common symbol override ifunc test 1b
+FAIL: Common symbol override ifunc test 1a
+FAIL: Common symbol override ifunc test 1b
 PASS: ld-ifunc/ifunc-1-local-x86
 PASS: ld-ifunc/ifunc-1-x86
 PASS: ld-ifunc/ifunc-10-i386
@@ -1067,8 +1067,10 @@ Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-xtensa/xtensa.exp ...
 
 		=== ld Summary ===
 
-# of expected passes		907
-# of expected failures		8
+# of expected passes		895
+# of unexpected failures	2
+# of unexpected successes	6
+# of expected failures		12
 # of untested testcases		6
 [...]/tschwinge/Paul_Desmond.build/ld/ld-new 2.23.52.20130618
 
