diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/binutils.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/binutils.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/binutils.sum	2012-08-29 22:06:18.503669945 +0200
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/binutils.sum	2012-08-29 22:06:26.647670202 +0200
@@ -1,5 +1,5 @@
-Test Run By thomas on Wed Aug 29 21:20:49 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Wed Aug 29 21:41:23 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== binutils tests ===
 
diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/gas.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/gas.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/gas.sum	2012-08-29 22:06:18.515669941 +0200
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/gas.sum	2012-08-29 22:06:26.663670198 +0200
@@ -1,5 +1,5 @@
-Test Run By thomas on Wed Aug 29 21:20:55 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Wed Aug 29 21:41:52 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== gas tests ===
 
diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/ld.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/ld.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/ld.sum	2012-08-29 22:06:18.531669942 +0200
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/ld.sum	2012-08-29 22:06:26.667670200 +0200
@@ -1,5 +1,5 @@
-Test Run By thomas on Wed Aug 29 21:21:19 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Wed Aug 29 21:43:43 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== ld tests ===
 
@@ -215,10 +215,10 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: PIE init array
 PASS: PIE fini array
 PASS: PIE init array mixed
-PASS: static preinit array
-PASS: static init array
-PASS: static fini array
-PASS: static init array mixed
+XFAIL: static preinit array
+XFAIL: static init array
+XFAIL: static fini array
+XFAIL: static init array mixed
 Running [...]/hurd/master/ld/testsuite/ld-elf/exclude.exp ...
 PASS: ld link shared library
 PASS: ld export symbols from archive
@@ -476,8 +476,8 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: ELF DSO weak func last DSO
 PASS: ELF weak func first
 PASS: ELF weak func last
-PASS: ELF weak func first DSO
-PASS: ELF weak func last DSO
+XFAIL: ELF weak func first DSO
+XFAIL: ELF weak func last DSO
 PASS: ELF DSO weak data first
 PASS: ELF DSO weak data last
 PASS: ELF DSO weak data first DSO
@@ -488,10 +488,10 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: ELF weak data last
 PASS: ELF weak data first common
 PASS: ELF weak data last common
-PASS: ELF weak data first DSO
-PASS: ELF weak data last DSO
-PASS: ELF weak data first DSO common
-PASS: ELF weak data last DSO common
+XFAIL: ELF weak data first DSO
+XFAIL: ELF weak data last DSO
+XFAIL: ELF weak data first DSO common
+XFAIL: ELF weak data last DSO common
 PASS: ELF DSO small bar (size)
 PASS: ELF DSO foo with small bar (size)
 PASS: ELF DSO big bar (size)
@@ -835,8 +835,8 @@ Running [...]/hurd/master/ld/testsuite/l
 
 		=== ld Summary ===
 
-# of expected passes		689
-# of expected failures		8
+# of expected passes		679
+# of expected failures		18
 # of untested testcases		6
 [...]/hurd/master.build/ld/ld-new 2.22.52.20120607
 
