diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/binutils/binutils.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/binutils/binutils.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/binutils/binutils.sum	2012-12-16 13:44:21.168215105 +0100
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/binutils/binutils.sum	2012-12-16 13:44:35.748215566 +0100
@@ -1,5 +1,5 @@
-Test Run By thomas on Sun Dec 16 09:49:36 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Sun Dec 16 10:41:46 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== binutils tests ===
 
diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/gas/testsuite/gas.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/gas/testsuite/gas.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/gas/testsuite/gas.sum	2012-12-16 13:44:21.184215105 +0100
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/gas/testsuite/gas.sum	2012-12-16 13:44:35.764215565 +0100
@@ -1,5 +1,5 @@
-Test Run By thomas on Sun Dec 16 09:49:42 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Sun Dec 16 10:42:14 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== gas tests ===
 
@@ -393,7 +393,8 @@ Running [...]/tschwinge/Paul_Desmond/gas
 PASS: i386 string-bad
 PASS: i386 reg-bad
 PASS: i386 space1
-PASS: gas/i386/rept
+WARNING: program timed out.
+FAIL: gas/i386/rept
 PASS: i386 list-1
 PASS: i386 list-2
 PASS: i386 list-3
@@ -510,6 +511,7 @@ Running [...]/tschwinge/Paul_Desmond/gas
 
 		=== gas Summary ===
 
-# of expected passes		379
+# of expected passes		378
+# of unexpected failures	1
 ../as-new 2.23.51.20121216
 
diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/ld/ld.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/ld/ld.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/ld/ld.sum	2012-12-16 13:44:21.184215105 +0100
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/ld/ld.sum	2012-12-16 13:44:35.764215565 +0100
@@ -1,5 +1,5 @@
-Test Run By thomas on Sun Dec 16 09:51:44 2012
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Sun Dec 16 11:12:25 2012
+Native configuration is i686-unknown-gnu0.3
 
 		=== ld tests ===
 
@@ -233,10 +233,10 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 PASS: PIE fini array
 PASS: PIE init array mixed
 PASS: PIE PR ld/14525
-PASS: static preinit array
-PASS: static init array
-PASS: static fini array
-PASS: static init array mixed
+XFAIL: static preinit array
+XFAIL: static init array
+XFAIL: static fini array
+XFAIL: static init array mixed
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-elf/exclude.exp ...
 PASS: ld link shared library
 PASS: ld export symbols from archive
@@ -536,8 +536,8 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 PASS: ELF DSO weak func last DSO
 PASS: ELF weak func first
 PASS: ELF weak func last
-PASS: ELF weak func first DSO
-PASS: ELF weak func last DSO
+XFAIL: ELF weak func first DSO
+XFAIL: ELF weak func last DSO
 PASS: ELF DSO weak data first
 PASS: ELF DSO weak data last
 PASS: ELF DSO weak data first DSO
@@ -548,10 +548,10 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 PASS: ELF weak data last
 PASS: ELF weak data first common
 PASS: ELF weak data last common
-PASS: ELF weak data first DSO
-PASS: ELF weak data last DSO
-PASS: ELF weak data first DSO common
-PASS: ELF weak data last DSO common
+XFAIL: ELF weak data first DSO
+XFAIL: ELF weak data last DSO
+XFAIL: ELF weak data first DSO common
+XFAIL: ELF weak data last DSO common
 PASS: ELF DSO small bar (size)
 PASS: ELF DSO foo with small bar (size)
 PASS: ELF DSO big bar (size)
@@ -647,8 +647,8 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ifunc/ifunc.exp ...
 PASS: Building ifunc binaries
 PASS: Checking ifunc binaries
-PASS: Common symbol override ifunc test 1a
-PASS: Common symbol override ifunc test 1b
+FAIL: Common symbol override ifunc test 1a
+FAIL: Common symbol override ifunc test 1b
 PASS: ld-ifunc/ifunc-1-local-x86
 PASS: ld-ifunc/ifunc-1-x86
 PASS: ld-ifunc/ifunc-10-i386
@@ -992,9 +992,9 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 
 		=== ld Summary ===
 
-# of expected passes		835
-# of unexpected failures	1
-# of expected failures		8
+# of expected passes		823
+# of unexpected failures	3
+# of expected failures		18
 # of untested testcases		6
 [...]/tschwinge/Paul_Desmond.build/ld/ld-new 2.23.51.20121216
 
