diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/binutils/binutils.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/binutils/binutils.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/binutils/binutils.sum	2013-02-08 14:36:30.185840184 +0100
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/binutils/binutils.sum	2013-02-08 14:36:43.085841347 +0100
@@ -1,5 +1,5 @@
-Test Run By thomas on Fri Feb  8 12:49:26 2013
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Fri Feb  8 13:40:48 2013
+Native configuration is i686-unknown-gnu0.3
 
 		=== binutils tests ===
 
diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/gas/testsuite/gas.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/gas/testsuite/gas.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/gas/testsuite/gas.sum	2013-02-08 14:36:30.185840184 +0100
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/gas/testsuite/gas.sum	2013-02-08 14:36:43.097841299 +0100
@@ -1,5 +1,5 @@
-Test Run By thomas on Fri Feb  8 12:49:32 2013
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Fri Feb  8 13:41:16 2013
+Native configuration is i686-unknown-gnu0.3
 
 		=== gas tests ===
 
@@ -397,7 +397,8 @@ Running [...]/tschwinge/Paul_Desmond/gas
 PASS: i386 string-bad
 PASS: i386 reg-bad
 PASS: i386 space1
-PASS: gas/i386/rept
+WARNING: program timed out.
+FAIL: gas/i386/rept
 PASS: i386 list-1
 PASS: i386 list-2
 PASS: i386 list-3
@@ -517,6 +518,7 @@ Running [...]/tschwinge/Paul_Desmond/gas
 
 		=== gas Summary ===
 
-# of expected passes		383
+# of expected passes		382
+# of unexpected failures	1
 ../as-new 2.23.51.20130208
 
diff -Nrwu -F '^Running' toolchain/logs/binutils/kepler.SCHWINGE/test/ld/ld.sum toolchain/logs/binutils/coulomb.SCHWINGE/test/ld/ld.sum
--- toolchain/logs/binutils/kepler.SCHWINGE/test/ld/ld.sum	2013-02-08 14:36:30.189840310 +0100
+++ toolchain/logs/binutils/coulomb.SCHWINGE/test/ld/ld.sum	2013-02-08 14:36:43.097841299 +0100
@@ -1,5 +1,5 @@
-Test Run By thomas on Fri Feb  8 12:51:34 2013
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Fri Feb  8 14:11:04 2013
+Native configuration is i686-unknown-gnu0.3
 
 		=== ld tests ===
 
@@ -242,10 +242,10 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 PASS: PIE fini array
 PASS: PIE init array mixed
 PASS: PIE PR ld/14525
-PASS: static preinit array
-PASS: static init array
-PASS: static fini array
-PASS: static init array mixed
+XFAIL: static preinit array
+XFAIL: static init array
+XFAIL: static fini array
+XFAIL: static init array mixed
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-elf/exclude.exp ...
 PASS: ld link shared library
 PASS: ld export symbols from archive
@@ -545,8 +545,8 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 PASS: ELF DSO weak func last DSO
 PASS: ELF weak func first
 PASS: ELF weak func last
-PASS: ELF weak func first DSO
-PASS: ELF weak func last DSO
+XFAIL: ELF weak func first DSO
+XFAIL: ELF weak func last DSO
 PASS: ELF DSO weak data first
 PASS: ELF DSO weak data last
 PASS: ELF DSO weak data first DSO
@@ -557,10 +557,10 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 PASS: ELF weak data last
 PASS: ELF weak data first common
 PASS: ELF weak data last common
-PASS: ELF weak data first DSO
-PASS: ELF weak data last DSO
-PASS: ELF weak data first DSO common
-PASS: ELF weak data last DSO common
+XFAIL: ELF weak data first DSO
+XFAIL: ELF weak data last DSO
+XFAIL: ELF weak data first DSO common
+XFAIL: ELF weak data last DSO common
 PASS: ELF DSO small bar (size)
 PASS: ELF DSO foo with small bar (size)
 PASS: ELF DSO big bar (size)
@@ -656,8 +656,8 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 Running [...]/tschwinge/Paul_Desmond/ld/testsuite/ld-ifunc/ifunc.exp ...
 PASS: Building ifunc binaries
 PASS: Checking ifunc binaries
-PASS: Common symbol override ifunc test 1a
-PASS: Common symbol override ifunc test 1b
+FAIL: Common symbol override ifunc test 1a
+FAIL: Common symbol override ifunc test 1b
 PASS: ld-ifunc/ifunc-1-local-x86
 PASS: ld-ifunc/ifunc-1-x86
 PASS: ld-ifunc/ifunc-10-i386
@@ -1039,9 +1039,9 @@ Running [...]/tschwinge/Paul_Desmond/ld/
 
 		=== ld Summary ===
 
-# of expected passes		879
-# of unexpected failures	1
-# of expected failures		8
+# of expected passes		867
+# of unexpected failures	3
+# of expected failures		18
 # of untested testcases		6
 [...]/tschwinge/Paul_Desmond.build/ld/ld-new 2.23.51.20130208
 
