--- /dev/fd/63	2011-06-19 18:21:05.226518816 +0200
+++ /dev/fd/62	2011-06-19 18:21:05.226518816 +0200
@@ -276,12 +276,12 @@
 checking for sys/sysinfo.h... yes
 checking for machine/hal_sysinfo.h... no
 checking for sys/table.h... no
-checking for sys/sysctl.h... yes
+checking for sys/sysctl.h... no
 checking for sys/systemcfg.h... no
 checking for stdint.h... (cached) yes
 checking for stdio_ext.h... yes
 checking for process.h... no
-checking for sys/prctl.h... yes
+checking for sys/prctl.h... no
 checking for sys/wait.h that is POSIX.1 compatible... yes
 checking whether time.h and sys/time.h may both be included... yes
 checking whether errno must be declared... no
@@ -351,13 +351,13 @@
 checking for working fork... yes
 checking for working vfork... (cached) yes
 checking for _doprnt... no
-checking for sys_errlist... yes
-checking for sys_nerr... yes
+checking for sys_errlist... no
+checking for sys_nerr... no
 checking for sys_siglist... yes
 checking for external symbol _system_configuration... no
 checking for __fsetlocking... yes
 checking for canonicalize_file_name... yes
-checking for dup3... yes
+checking for dup3... no
 checking for getrusage... yes
 checking for getsysinfo... no
 checking for gettimeofday... (cached) yes
@@ -372,7 +372,7 @@
 checking for strerror... yes
 checking for strsignal... yes
 checking for sysconf... yes
-checking for sysctl... yes
+checking for sysctl... no
 checking for sysmp... no
 checking for table... no
 checking for times... yes
@@ -457,7 +457,7 @@
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -473,18 +473,18 @@
 checking if gcc-4.6 supports -fno-rtti -fno-exceptions... no
 checking for gcc-4.6 option to produce PIC... -fPIC -DPIC
 checking if gcc-4.6 PIC flag -fPIC -DPIC works... yes
-checking if gcc-4.6 static flag -static works... yes
+checking if gcc-4.6 static flag -static works... no
 checking if gcc-4.6 supports -c -o file.o... yes
 checking if gcc-4.6 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.6 linker (ld) supports shared libraries... yes
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.3 ld.so
 checking how to hardcode library paths into programs... immediate
 checking for shl_load... no
 checking for shl_load in -ldld... no
 checking for dlopen... no
 checking for dlopen in -ldl... yes
 checking whether a program can dlopen itself... yes
-checking whether a statically linked program can dlopen itself... no
+checking whether a statically linked program can dlopen itself... yes
 checking whether stripping libraries is possible... yes
 checking if libtool supports shared libraries... yes
 checking whether to build shared libraries... no
@@ -567,26 +567,26 @@
 checking sys/procfs.h usability... yes
 checking sys/procfs.h presence... yes
 checking for sys/procfs.h... yes
-checking for prstatus_t in sys/procfs.h... yes
+checking for prstatus_t in sys/procfs.h... no
 checking for prstatus32_t in sys/procfs.h... no
 checking for prstatus_t.pr_who in sys/procfs.h... no
 checking for prstatus32_t.pr_who in sys/procfs.h... no
-checking for pstatus_t in sys/procfs.h... no
+checking for pstatus_t in sys/procfs.h... yes
 checking for pxstatus_t in sys/procfs.h... no
 checking for pstatus32_t in sys/procfs.h... no
-checking for prpsinfo_t in sys/procfs.h... yes
-checking for prpsinfo_t.pr_pid in sys/procfs.h... yes
+checking for prpsinfo_t in sys/procfs.h... no
+checking for prpsinfo_t.pr_pid in sys/procfs.h... no
 checking for prpsinfo32_t in sys/procfs.h... no
 checking for prpsinfo32_t.pr_pid in sys/procfs.h... no
-checking for psinfo_t in sys/procfs.h... no
-checking for psinfo_t.pr_pid in sys/procfs.h... no
+checking for psinfo_t in sys/procfs.h... yes
+checking for psinfo_t.pr_pid in sys/procfs.h... yes
 checking for psinfo32_t in sys/procfs.h... no
 checking for psinfo32_t.pr_pid in sys/procfs.h... no
-checking for lwpstatus_t in sys/procfs.h... no
+checking for lwpstatus_t in sys/procfs.h... yes
 checking for lwpxstatus_t in sys/procfs.h... no
 checking for lwpstatus_t.pr_context in sys/procfs.h... no
-checking for lwpstatus_t.pr_reg in sys/procfs.h... no
-checking for lwpstatus_t.pr_fpreg in sys/procfs.h... no
+checking for lwpstatus_t.pr_reg in sys/procfs.h... yes
+checking for lwpstatus_t.pr_fpreg in sys/procfs.h... yes
 checking for win32_pstatus_t in sys/procfs.h... no
 checking linker --as-needed support... yes
 checking for cos in -lm... yes
@@ -601,7 +601,7 @@
 checking for unistd.h... (cached) yes
 checking for getpagesize... (cached) yes
 checking for working mmap... yes
-checking for madvise... yes
+checking for madvise... no
 checking for mprotect... yes
 configure: updating cache ./config.cache
 configure: creating ./config.status
@@ -1219,36 +1219,15 @@
 /bin/dash ./libtool --tag=CC   --mode=compile gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd    -I. -I../../master/bfd -I../../master/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/hurd/master.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT dwarf1.lo -MD -MP -MF .deps/dwarf1.Tpo -c -o dwarf1.lo ../../master/bfd/dwarf1.c
 libtool: compile:  gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd  -I. -I../../master/bfd -I../../master/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/hurd/master.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT dwarf1.lo -MD -MP -MF .deps/dwarf1.Tpo -c ../../master/bfd/dwarf1.c -o dwarf1.o
 mv -f .deps/dwarf1.Tpo .deps/dwarf1.Plo
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd    -I. -I../../master/bfd -I../../master/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/hurd/master.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT i386linux.lo -MD -MP -MF .deps/i386linux.Tpo -c -o i386linux.lo ../../master/bfd/i386linux.c
-libtool: compile:  gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd  -I. -I../../master/bfd -I../../master/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/hurd/master.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT i386linux.lo -MD -MP -MF .deps/i386linux.Tpo -c ../../master/bfd/i386linux.c -o i386linux.o
-mv -f .deps/i386linux.Tpo .deps/i386linux.Plo
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd    -I. -I../../master/bfd -I../../master/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/hurd/master.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT aout32.lo -MD -MP -MF .deps/aout32.Tpo -c -o aout32.lo ../../master/bfd/aout32.c
-libtool: compile:  gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd  -I. -I../../master/bfd -I../../master/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/hurd/master.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT aout32.lo -MD -MP -MF .deps/aout32.Tpo -c ../../master/bfd/aout32.c -o aout32.o
-mv -f .deps/aout32.Tpo .deps/aout32.Plo
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd    -I. -I../../master/bfd -I../../master/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/hurd/master.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT pei-i386.lo -MD -MP -MF .deps/pei-i386.Tpo -c -o pei-i386.lo ../../master/bfd/pei-i386.c
-libtool: compile:  gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd  -I. -I../../master/bfd -I../../master/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/hurd/master.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT pei-i386.lo -MD -MP -MF .deps/pei-i386.Tpo -c ../../master/bfd/pei-i386.c -o pei-i386.o
-mv -f .deps/pei-i386.Tpo .deps/pei-i386.Plo
-rm -f peigen.c
-sed -e s/XX/pe/g < ../../master/bfd/peXXigen.c > peigen.new
-mv -f peigen.new peigen.c
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd    -I. -I../../master/bfd -I../../master/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/hurd/master.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT peigen.lo -MD -MP -MF .deps/peigen.Tpo -c -o peigen.lo peigen.c
-libtool: compile:  gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd  -I. -I../../master/bfd -I../../master/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/hurd/master.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT peigen.lo -MD -MP -MF .deps/peigen.Tpo -c peigen.c -o peigen.o
-mv -f .deps/peigen.Tpo .deps/peigen.Plo
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd    -I. -I../../master/bfd -I../../master/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/hurd/master.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT cofflink.lo -MD -MP -MF .deps/cofflink.Tpo -c -o cofflink.lo ../../master/bfd/cofflink.c
-libtool: compile:  gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd  -I. -I../../master/bfd -I../../master/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/hurd/master.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT cofflink.lo -MD -MP -MF .deps/cofflink.Tpo -c ../../master/bfd/cofflink.c -o cofflink.o
-mv -f .deps/cofflink.Tpo .deps/cofflink.Plo
 /bin/dash ./libtool --tag=CC   --mode=compile gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd    -I. -I../../master/bfd -I../../master/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/hurd/master.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT elf32-gen.lo -MD -MP -MF .deps/elf32-gen.Tpo -c -o elf32-gen.lo ../../master/bfd/elf32-gen.c
 libtool: compile:  gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd  -I. -I../../master/bfd -I../../master/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/hurd/master.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT elf32-gen.lo -MD -MP -MF .deps/elf32-gen.Tpo -c ../../master/bfd/elf32-gen.c -o elf32-gen.o
 mv -f .deps/elf32-gen.Tpo .deps/elf32-gen.Plo
 /bin/dash ./libtool --tag=CC   --mode=compile gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd    -I. -I../../master/bfd -I../../master/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/hurd/master.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT cpu-i386.lo -MD -MP -MF .deps/cpu-i386.Tpo -c -o cpu-i386.lo ../../master/bfd/cpu-i386.c
 libtool: compile:  gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd  -I. -I../../master/bfd -I../../master/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/hurd/master.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT cpu-i386.lo -MD -MP -MF .deps/cpu-i386.Tpo -c ../../master/bfd/cpu-i386.c -o cpu-i386.o
 mv -f .deps/cpu-i386.Tpo .deps/cpu-i386.Plo
-/bin/dash ./libtool --tag=CC   --mode=compile gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd    -I. -I../../master/bfd -I../../master/bfd/../include  -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec  -DBINDIR='"[...]/hurd/master.build.install/bin"'  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT trad-core.lo -MD -MP -MF .deps/trad-core.Tpo -c -o trad-core.lo ../../master/bfd/trad-core.c
-libtool: compile:  gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/bfd  -I. -I../../master/bfd -I../../master/bfd/../include -DHAVE_bfd_elf32_i386_vec   -DHAVE_bfd_elf32_little_generic_vec -DHAVE_bfd_elf32_big_generic_vec -DBINDIR=\"[...]/hurd/master.build.install/bin\" -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT trad-core.lo -MD -MP -MF .deps/trad-core.Tpo -c ../../master/bfd/trad-core.c -o trad-core.o
-mv -f .deps/trad-core.Tpo .deps/trad-core.Plo
 rm -f tofiles
 f=""; \
-	for i in elf32-i386.lo elf-ifunc.lo elf-vxworks.lo elf32.lo elf.lo elflink.lo elf-attrs.lo elf-strtab.lo elf-eh-frame.lo dwarf1.lo i386linux.lo aout32.lo pei-i386.lo peigen.lo cofflink.lo elf32-gen.lo cpu-i386.lo trad-core.lo  ; do \
+	for i in elf32-i386.lo elf-ifunc.lo elf-vxworks.lo elf32.lo elf.lo elflink.lo elf-attrs.lo elf-strtab.lo elf-eh-frame.lo dwarf1.lo elf32-gen.lo cpu-i386.lo   ; do \
 	  case " $f " in \
 	    *" $i "*) ;; \
 	    *) f="$f $i" ;; \
@@ -1258,7 +1237,7 @@
 /bin/dash ../../master/bfd/../move-if-change tofiles ofiles
 touch stamp-ofiles
 /bin/dash ./libtool --tag=CC   --mode=link gcc-4.6 -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2  -release `cat libtool-soversion`   -o libbfd.la -rpath [...]/hurd/master.build.install/lib archive.lo archures.lo bfd.lo bfdio.lo bfdwin.lo cache.lo coffgen.lo corefile.lo format.lo init.lo libbfd.lo opncls.lo reloc.lo section.lo syms.lo targets.lo hash.lo linker.lo srec.lo binary.lo tekhex.lo ihex.lo stabs.lo stab-syms.lo merge.lo dwarf2.lo simple.lo compress.lo verilog.lo `cat ofiles`   -lz 
-libtool: link: ar rc .libs/libbfd.a  archive.o archures.o bfd.o bfdio.o bfdwin.o cache.o coffgen.o corefile.o format.o init.o libbfd.o opncls.o reloc.o section.o syms.o targets.o hash.o linker.o srec.o binary.o tekhex.o ihex.o stabs.o stab-syms.o merge.o dwarf2.o simple.o compress.o verilog.o elf32-i386.o elf-ifunc.o elf-vxworks.o elf32.o elf.o elflink.o elf-attrs.o elf-strtab.o elf-eh-frame.o dwarf1.o i386linux.o aout32.o pei-i386.o peigen.o cofflink.o elf32-gen.o cpu-i386.o trad-core.o
+libtool: link: ar rc .libs/libbfd.a  archive.o archures.o bfd.o bfdio.o bfdwin.o cache.o coffgen.o corefile.o format.o init.o libbfd.o opncls.o reloc.o section.o syms.o targets.o hash.o linker.o srec.o binary.o tekhex.o ihex.o stabs.o stab-syms.o merge.o dwarf2.o simple.o compress.o verilog.o elf32-i386.o elf-ifunc.o elf-vxworks.o elf32.o elf.o elflink.o elf-attrs.o elf-strtab.o elf-eh-frame.o dwarf1.o elf32-gen.o cpu-i386.o
 libtool: link: ranlib .libs/libbfd.a
 libtool: link: ( cd ".libs" && rm -f "libbfd.la" && ln -s "../libbfd.la" "libbfd.la" )
 libtooldir=`/bin/dash ./libtool --config | sed -n -e 's/^objdir=//p'`; \
@@ -1325,7 +1304,7 @@
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -1341,11 +1320,11 @@
 checking if gcc-4.6 supports -fno-rtti -fno-exceptions... no
 checking for gcc-4.6 option to produce PIC... -fPIC -DPIC
 checking if gcc-4.6 PIC flag -fPIC -DPIC works... yes
-checking if gcc-4.6 static flag -static works... yes
+checking if gcc-4.6 static flag -static works... no
 checking if gcc-4.6 supports -c -o file.o... yes
 checking if gcc-4.6 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.6 linker (ld) supports shared libraries... yes
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.3 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether stripping libraries is possible... yes
 checking if libtool supports shared libraries... yes
@@ -1500,7 +1479,7 @@
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -1516,12 +1495,12 @@
 checking if gcc-4.6 supports -fno-rtti -fno-exceptions... no
 checking for gcc-4.6 option to produce PIC... -fPIC -DPIC
 checking if gcc-4.6 PIC flag -fPIC -DPIC works... yes
-checking if gcc-4.6 static flag -static works... yes
+checking if gcc-4.6 static flag -static works... no
 checking if gcc-4.6 supports -c -o file.o... yes
 checking if gcc-4.6 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.6 linker (ld) supports shared libraries... yes
 checking whether -lc should be explicitly linked in... no
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.3 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether stripping libraries is possible... yes
 checking if libtool supports shared libraries... yes
@@ -1967,7 +1946,7 @@
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -1983,12 +1962,12 @@
 checking if gcc-4.6 supports -fno-rtti -fno-exceptions... no
 checking for gcc-4.6 option to produce PIC... -fPIC -DPIC
 checking if gcc-4.6 PIC flag -fPIC -DPIC works... yes
-checking if gcc-4.6 static flag -static works... yes
+checking if gcc-4.6 static flag -static works... no
 checking if gcc-4.6 supports -c -o file.o... yes
 checking if gcc-4.6 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.6 linker (ld) supports shared libraries... yes
 checking whether -lc should be explicitly linked in... no
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.3 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether stripping libraries is possible... yes
 checking if libtool supports shared libraries... yes
@@ -2221,7 +2200,7 @@
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -2237,12 +2216,12 @@
 checking if gcc-4.6 supports -fno-rtti -fno-exceptions... no
 checking for gcc-4.6 option to produce PIC... -fPIC -DPIC
 checking if gcc-4.6 PIC flag -fPIC -DPIC works... yes
-checking if gcc-4.6 static flag -static works... yes
+checking if gcc-4.6 static flag -static works... no
 checking if gcc-4.6 supports -c -o file.o... yes
 checking if gcc-4.6 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.6 linker (ld) supports shared libraries... yes
 checking whether -lc should be explicitly linked in... no
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.3 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether stripping libraries is possible... yes
 checking if libtool supports shared libraries... yes
@@ -2461,7 +2440,7 @@
 checking for BSD- or MS-compatible name lister (nm)... nm
 checking the name lister (nm) interface... BSD nm
 checking whether ln -s works... yes
-checking the maximum length of command line arguments... 805306365
+checking the maximum length of command line arguments... -1
 checking whether the shell understands some XSI constructs... yes
 checking whether the shell understands "+="... no
 checking for ld option to reload object files... -r
@@ -2477,12 +2456,12 @@
 checking if gcc-4.6 supports -fno-rtti -fno-exceptions... no
 checking for gcc-4.6 option to produce PIC... -fPIC -DPIC
 checking if gcc-4.6 PIC flag -fPIC -DPIC works... yes
-checking if gcc-4.6 static flag -static works... yes
+checking if gcc-4.6 static flag -static works... no
 checking if gcc-4.6 supports -c -o file.o... yes
 checking if gcc-4.6 supports -c -o file.o... (cached) yes
 checking whether the gcc-4.6 linker (ld) supports shared libraries... yes
 checking whether -lc should be explicitly linked in... no
-checking dynamic linker characteristics... GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.3 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether stripping libraries is possible... yes
 checking if libtool supports shared libraries... yes
@@ -2494,11 +2473,11 @@
 checking whether the g++-4.6 linker (ld) supports shared libraries... yes
 checking for g++-4.6 option to produce PIC... -fPIC -DPIC
 checking if g++-4.6 PIC flag -fPIC -DPIC works... yes
-checking if g++-4.6 static flag -static works... yes
+checking if g++-4.6 static flag -static works... no
 checking if g++-4.6 supports -c -o file.o... yes
 checking if g++-4.6 supports -c -o file.o... (cached) yes
 checking whether the g++-4.6 linker (ld) supports shared libraries... yes
-checking dynamic linker characteristics... (cached) GNU/Linux ld.so
+checking dynamic linker characteristics... gnu0.3 ld.so
 checking how to hardcode library paths into programs... immediate
 checking whether NLS is requested... yes
 checking for catalogs to be installed...  bg da es fi fr ga id ja sv tr vi zh_CN zh_TW
@@ -2578,13 +2557,13 @@
 /bin/dash ../../master/ld/../ylwrap ../../master/ld/ldgram.y y.tab.c ldgram.c y.tab.h ldgram.h y.output ldgram.output -- bison -y -d 
 updating ldgram.h
 (echo "/* This file is automatically generated.  DO NOT EDIT! */";\
-	for f in `echo " " eelf_i386.o ei386linux.o eelf32_x86_64.o "" \
+	for f in `echo " " eelf_i386.o "" \
 	 | sed -e 's/ e/ ld/g' -e 's/ ld/ /g' -e 's/[.]o//g'`; do \
 	    echo "extern ld_emulation_xfer_type ld_${f}_emulation;"; \
 	done;\
 	echo "";\
 	echo "#define EMULATION_LIST \\";\
-	for f in `echo " " eelf_i386.o ei386linux.o eelf32_x86_64.o "" \
+	for f in `echo " " eelf_i386.o "" \
 	 | sed -e 's/ e/ ld/g' -e 's/ ld/ /g' -e 's/[.]o//g'`; do \
 	    echo "  &ld_${f}_emulation, \\"; \
 	done;\
@@ -2696,17 +2675,11 @@
 gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/ld  -I. -I../../master/ld -I../bfd -I../../master/ld/../bfd -I../../master/ld/../include  -g -O2 -DENABLE_PLUGINS -DLOCALEDIR="\"[...]/hurd/master.build.install/share/locale\""  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT plugin.o -MD -MP -MF .deps/plugin.Tpo -c -o plugin.o ../../master/ld/plugin.c
 mv -f .deps/plugin.Tpo .deps/plugin.Po
 cp ../../master/ld/emultempl/astring.sed stringify.sed
-LIB_PATH='' /bin/dash ../../master/ld/genscripts.sh "../../master/ld" "[...]/hurd/master.build.install/lib" "[...]/hurd/master.build.install" "[...]/hurd/master.build.install" [ARCH] [ARCH] [ARCH] "elf_i386 elf32_x86_64" "/usr/local/lib /lib /usr/lib" no yes elf_i386 "[ARCH]"
+LIB_PATH='' /bin/dash ../../master/ld/genscripts.sh "../../master/ld" "[...]/hurd/master.build.install/lib" "[...]/hurd/master.build.install" "[...]/hurd/master.build.install" [ARCH] [ARCH] [ARCH] "elf_i386" "/usr/local/lib /lib /usr/lib" no yes elf_i386 "[ARCH]"
 gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/ld  -I. -I../../master/ld -I../bfd -I../../master/ld/../bfd -I../../master/ld/../include  -g -O2 -DENABLE_PLUGINS -DLOCALEDIR="\"[...]/hurd/master.build.install/share/locale\""  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT eelf_i386.o -MD -MP -MF .deps/eelf_i386.Tpo -c -o eelf_i386.o eelf_i386.c
 mv -f .deps/eelf_i386.Tpo .deps/eelf_i386.Po
-LIB_PATH='' /bin/dash ../../master/ld/genscripts.sh "../../master/ld" "[...]/hurd/master.build.install/lib" "[...]/hurd/master.build.install" "[...]/hurd/master.build.install" [ARCH] [ARCH] [ARCH] "elf_i386 elf32_x86_64" "/usr/local/lib /lib /usr/lib" no yes i386linux "[ARCH]aout"
-gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/ld  -I. -I../../master/ld -I../bfd -I../../master/ld/../bfd -I../../master/ld/../include  -g -O2 -DENABLE_PLUGINS -DLOCALEDIR="\"[...]/hurd/master.build.install/share/locale\""  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT ei386linux.o -MD -MP -MF .deps/ei386linux.Tpo -c -o ei386linux.o ei386linux.c
-mv -f .deps/ei386linux.Tpo .deps/ei386linux.Po
-LIB_PATH='' /bin/dash ../../master/ld/genscripts.sh "../../master/ld" "[...]/hurd/master.build.install/lib" "[...]/hurd/master.build.install" "[...]/hurd/master.build.install" [ARCH] [ARCH] [ARCH] "elf_i386 elf32_x86_64" "/usr/local/lib /lib /usr/lib" no yes elf32_x86_64 "[ARCH]"
-gcc-4.6 -DHAVE_CONFIG_H -I. -I../../master/ld  -I. -I../../master/ld -I../bfd -I../../master/ld/../bfd -I../../master/ld/../include  -g -O2 -DENABLE_PLUGINS -DLOCALEDIR="\"[...]/hurd/master.build.install/share/locale\""  -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -MT eelf32_x86_64.o -MD -MP -MF .deps/eelf32_x86_64.Tpo -c -o eelf32_x86_64.o eelf32_x86_64.c
-mv -f .deps/eelf32_x86_64.Tpo .deps/eelf32_x86_64.Po
-/bin/dash ./libtool --tag=CC   --mode=link gcc-4.6 -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2   -o ld-new ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o ldwrite.o ldexp.o ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o plugin.o eelf_i386.o ei386linux.o eelf32_x86_64.o  ../bfd/libbfd.la ../libiberty/libiberty.a  -lz -ldl 
-libtool: link: gcc-4.6 -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -o ld-new ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o ldwrite.o ldexp.o ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o plugin.o eelf_i386.o ei386linux.o eelf32_x86_64.o  ../bfd/.libs/libbfd.a ../libiberty/libiberty.a -lz -ldl
+/bin/dash ./libtool --tag=CC   --mode=link gcc-4.6 -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2   -o ld-new ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o ldwrite.o ldexp.o ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o plugin.o eelf_i386.o  ../bfd/libbfd.la ../libiberty/libiberty.a  -lz -ldl 
+libtool: link: gcc-4.6 -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Werror -g -O2 -o ld-new ldgram.o ldlex-wrapper.o lexsup.o ldlang.o mri.o ldctor.o ldmain.o ldwrite.o ldexp.o ldemul.o ldver.o ldmisc.o ldfile.o ldcref.o plugin.o eelf_i386.o  ../bfd/.libs/libbfd.a ../libiberty/libiberty.a -lz -ldl
 touch ld.1
 perl ../../master/ld/../etc/texi2pod.pl -I ../../master/ld -I ../../master/ld/../bfd/doc -I ../bfd/doc -I ../../master/ld/../libiberty -Dman < ../../master/ld/ld.texinfo > ld.pod
 (pod2man --center="GNU Development Tools" --release="binutils-2.21.52" --section=1 ld.pod | \
