--- open_issues/binutils/sum_linux	2011-09-04 15:01:50.000000000 +0200
+++ open_issues/binutils/sum_hurd	2011-09-04 15:01:57.000000000 +0200
@@ -1,5 +1,5 @@
-Test Run By thomas on Sun Sep  4 14:45:17 2011
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Sun Sep  4 14:47:58 2011
+Native configuration is i686-unknown-gnu0.3
 
 		=== binutils tests ===
 
@@ -120,8 +120,8 @@ Running [...]/hurd/master/binutils/tests
 
 # of expected passes		88
 # of unsupported tests		3
-Test Run By thomas on Sun Sep  4 14:45:54 2011
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Sun Sep  4 14:50:18 2011
+Native configuration is i686-unknown-gnu0.3
 
 		=== ld tests ===
 
@@ -325,10 +325,10 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: PIE init array
 PASS: PIE fini array
 PASS: PIE init array mixed
-PASS: static preinit array
-PASS: static init array
-PASS: static fini array
-PASS: static init array mixed
+XFAIL: static preinit array
+XFAIL: static init array
+XFAIL: static fini array
+XFAIL: static init array mixed
 Running [...]/hurd/master/ld/testsuite/ld-elf/exclude.exp ...
 PASS: ld link shared library
 PASS: ld export symbols from archive
@@ -582,8 +582,8 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: ELF DSO weak func last DSO
 PASS: ELF weak func first
 PASS: ELF weak func last
-PASS: ELF weak func first DSO
-PASS: ELF weak func last DSO
+XFAIL: ELF weak func first DSO
+XFAIL: ELF weak func last DSO
 PASS: ELF DSO weak data first
 PASS: ELF DSO weak data last
 PASS: ELF DSO weak data first DSO
@@ -594,10 +594,10 @@ Running [...]/hurd/master/ld/testsuite/l
 PASS: ELF weak data last
 PASS: ELF weak data first common
 PASS: ELF weak data last common
-PASS: ELF weak data first DSO
-PASS: ELF weak data last DSO
-PASS: ELF weak data first DSO common
-PASS: ELF weak data last DSO common
+XFAIL: ELF weak data first DSO
+XFAIL: ELF weak data last DSO
+XFAIL: ELF weak data first DSO common
+XFAIL: ELF weak data last DSO common
 PASS: ELF DSO small bar (size)
 PASS: ELF DSO foo with small bar (size)
 PASS: ELF DSO big bar (size)
@@ -928,13 +928,13 @@ Running [...]/hurd/master/ld/testsuite/l
 
 		=== ld Summary ===
 
-# of expected passes		662
-# of expected failures		8
+# of expected passes		652
+# of expected failures		18
 # of untested testcases		6
 [...]/hurd/master.build/ld/ld-new 2.21.53.20110904
 
-Test Run By thomas on Sun Sep  4 14:45:29 2011
-Native configuration is i686-pc-linux-gnu
+Test Run By thomas on Sun Sep  4 14:48:36 2011
+Native configuration is i686-unknown-gnu0.3
 
 		=== gas tests ===
 
